From: Stained Glass OS <ke7oxh@gmail.com>
Subject: [PATCH] msvcp: Implement _Yarn<wchar_t/unsigned short>, _Facet_base and _UShinit members.

Class members that were spec stubs: _Yarn<wchar_t>'s copy and string
constructors, assignment, c_str and empty, and all of _Yarn<unsigned
short> (the same layout, the wchar_t functions) in msvcp140; _Facet_base's
constructors (they set its vtable), destructor and assignment (msvcp110
and later); _UShinit, the unsigned short streams' counterpart of _Winit
with its own count (flushes ucout, ucerr and uclog when the last one goes;
msvcp100 and later); and the assignments of the data-less _Iosb<int>,
_Init_locks and _Crt_new_delete. msvcp_win and msvcp120_app forward.
---
 dlls/msvcp100/msvcp100.spec         |  30 +++----
 dlls/msvcp110/msvcp110.spec         |  54 ++++++------
 dlls/msvcp120/msvcp120.spec         |  54 ++++++------
 dlls/msvcp120_app/msvcp120_app.spec |  54 ++++++------
 dlls/msvcp140/msvcp140.spec         | 170 ++++++++++++++++++------------------
 dlls/msvcp90/ios.c                  |  49 +++++++++++
 dlls/msvcp90/locale.c               |  32 +++++++
 dlls/msvcp90/string.c               |  36 ++++++++
 dlls/msvcp_win/msvcp_win.spec       | 166 +++++++++++++++++------------------
 9 files changed, 381 insertions(+), 264 deletions(-)

diff --git a/dlls/msvcp100/msvcp100.spec b/dlls/msvcp100/msvcp100.spec
index 0b22a5c..ecd4ce9 100644
--- a/dlls/msvcp100/msvcp100.spec
+++ b/dlls/msvcp100/msvcp100.spec
@@ -310,9 +310,9 @@
 @ cdecl -arch=arm ??0_Timevec@std@@QAA@PAX@Z(ptr ptr) _Timevec_ctor_timeptr
 @ thiscall -arch=i386 ??0_Timevec@std@@QAE@PAX@Z(ptr ptr) _Timevec_ctor_timeptr
 @ cdecl -arch=win64 ??0_Timevec@std@@QEAA@PEAX@Z(ptr ptr) _Timevec_ctor_timeptr
-@ stub -arch=arm ??0_UShinit@std@@QAA@XZ
-@ stub -arch=i386 ??0_UShinit@std@@QAE@XZ
-@ stub -arch=win64 ??0_UShinit@std@@QEAA@XZ
+@ cdecl -arch=arm ??0_UShinit@std@@QAA@XZ(ptr) _UShinit_ctor
+@ thiscall -arch=i386 ??0_UShinit@std@@QAE@XZ(ptr) _UShinit_ctor
+@ cdecl -arch=win64 ??0_UShinit@std@@QEAA@XZ(ptr) _UShinit_ctor
 @ cdecl -arch=arm ??0_Winit@std@@QAA@XZ(ptr) _Winit_ctor
 @ thiscall -arch=i386 ??0_Winit@std@@QAE@XZ(ptr) _Winit_ctor
 @ cdecl -arch=win64 ??0_Winit@std@@QEAA@XZ(ptr) _Winit_ctor
@@ -481,9 +481,9 @@
 @ cdecl -arch=arm ??1_Timevec@std@@QAA@XZ(ptr) _Timevec_dtor
 @ thiscall -arch=i386 ??1_Timevec@std@@QAE@XZ(ptr) _Timevec_dtor
 @ cdecl -arch=win64 ??1_Timevec@std@@QEAA@XZ(ptr) _Timevec_dtor
-@ stub -arch=arm ??1_UShinit@std@@QAA@XZ
-@ stub -arch=i386 ??1_UShinit@std@@QAE@XZ
-@ stub -arch=win64 ??1_UShinit@std@@QEAA@XZ
+@ cdecl -arch=arm ??1_UShinit@std@@QAA@XZ(ptr) _UShinit_dtor
+@ thiscall -arch=i386 ??1_UShinit@std@@QAE@XZ(ptr) _UShinit_dtor
+@ cdecl -arch=win64 ??1_UShinit@std@@QEAA@XZ(ptr) _UShinit_dtor
 @ cdecl -arch=arm ??1_Winit@std@@QAA@XZ(ptr) _Winit_dtor
 @ thiscall -arch=i386 ??1_Winit@std@@QAE@XZ(ptr) _Winit_dtor
 @ cdecl -arch=win64 ??1_Winit@std@@QEAA@XZ(ptr) _Winit_dtor
@@ -505,9 +505,9 @@
 @ cdecl -arch=arm ??1time_base@std@@UAA@XZ(ptr) time_base_dtor
 @ thiscall -arch=i386 ??1time_base@std@@UAE@XZ(ptr) time_base_dtor
 @ cdecl -arch=win64 ??1time_base@std@@UEAA@XZ(ptr) time_base_dtor
-@ stub -arch=arm ??4?$_Iosb@H@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Iosb@H@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Iosb@H@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4?$_Iosb@H@std@@QAAAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ thiscall -arch=i386 ??4?$_Iosb@H@std@@QAEAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=win64 ??4?$_Iosb@H@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) empty_class_op_assign
 @ cdecl -arch=arm ??4?$_Yarn@D@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Yarn_char_op_assign
 @ thiscall -arch=i386 ??4?$_Yarn@D@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Yarn_char_op_assign
 @ cdecl -arch=win64 ??4?$_Yarn@D@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Yarn_char_op_assign
@@ -607,9 +607,9 @@
 @ cdecl -arch=arm ??4_Container_base12@std@@QAAAAU01@ABU01@@Z(ptr ptr) _Container_base12_op_assign
 @ thiscall -arch=i386 ??4_Container_base12@std@@QAEAAU01@ABU01@@Z(ptr ptr) _Container_base12_op_assign
 @ cdecl -arch=win64 ??4_Container_base12@std@@QEAAAEAU01@AEBU01@@Z(ptr ptr) _Container_base12_op_assign
-@ stub -arch=arm ??4_Init_locks@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_Init_locks@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_Init_locks@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4_Init_locks@std@@QAAAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ thiscall -arch=i386 ??4_Init_locks@std@@QAEAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=win64 ??4_Init_locks@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) empty_class_op_assign
 @ cdecl -arch=arm ??4_Num_base@std@@QAAAAU01@ABU01@@Z(ptr ptr) std_Num_base_op_assign
 @ thiscall -arch=i386 ??4_Num_base@std@@QAEAAU01@ABU01@@Z(ptr ptr) std_Num_base_op_assign
 @ cdecl -arch=win64 ??4_Num_base@std@@QEAAAEAU01@AEBU01@@Z(ptr ptr) std_Num_base_op_assign
@@ -622,9 +622,9 @@
 @ cdecl -arch=arm ??4_Timevec@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Timevec_op_assign
 @ thiscall -arch=i386 ??4_Timevec@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Timevec_op_assign
 @ cdecl -arch=win64 ??4_Timevec@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Timevec_op_assign
-@ stub -arch=arm ??4_UShinit@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_UShinit@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_UShinit@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4_UShinit@std@@QAAAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ thiscall -arch=i386 ??4_UShinit@std@@QAEAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=win64 ??4_UShinit@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) empty_class_op_assign
 @ cdecl -arch=arm ??4_Winit@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Winit_op_assign
 @ thiscall -arch=i386 ??4_Winit@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Winit_op_assign
 @ cdecl -arch=win64 ??4_Winit@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Winit_op_assign
diff --git a/dlls/msvcp110/msvcp110.spec b/dlls/msvcp110/msvcp110.spec
index 63a30de..cf61ecf 100644
--- a/dlls/msvcp110/msvcp110.spec
+++ b/dlls/msvcp110/msvcp110.spec
@@ -283,12 +283,12 @@
 @ cdecl -arch=arm ??0_Container_base12@std@@QAA@XZ(ptr) _Container_base12_ctor
 @ thiscall -arch=i386 ??0_Container_base12@std@@QAE@XZ(ptr) _Container_base12_ctor
 @ cdecl -arch=win64 ??0_Container_base12@std@@QEAA@XZ(ptr) _Container_base12_ctor
-@ stub -arch=arm ??0_Facet_base@std@@QAA@ABV01@@Z
-@ stub -arch=i386 ??0_Facet_base@std@@QAE@ABV01@@Z
-@ stub -arch=win64 ??0_Facet_base@std@@QEAA@AEBV01@@Z
-@ stub -arch=arm ??0_Facet_base@std@@QAA@XZ
-@ stub -arch=i386 ??0_Facet_base@std@@QAE@XZ
-@ stub -arch=win64 ??0_Facet_base@std@@QEAA@XZ
+@ cdecl -arch=arm ??0_Facet_base@std@@QAA@ABV01@@Z(ptr ptr) _Facet_base_ctor
+@ thiscall -arch=i386 ??0_Facet_base@std@@QAE@ABV01@@Z(ptr ptr) _Facet_base_ctor
+@ cdecl -arch=win64 ??0_Facet_base@std@@QEAA@AEBV01@@Z(ptr ptr) _Facet_base_ctor
+@ cdecl -arch=arm ??0_Facet_base@std@@QAA@XZ(ptr ptr) _Facet_base_ctor
+@ thiscall -arch=i386 ??0_Facet_base@std@@QAE@XZ(ptr ptr) _Facet_base_ctor
+@ cdecl -arch=win64 ??0_Facet_base@std@@QEAA@XZ(ptr ptr) _Facet_base_ctor
 @ cdecl -arch=arm ??0_Init_locks@std@@QAA@XZ(ptr) _Init_locks_ctor
 @ thiscall -arch=i386 ??0_Init_locks@std@@QAE@XZ(ptr) _Init_locks_ctor
 @ cdecl -arch=win64 ??0_Init_locks@std@@QEAA@XZ(ptr) _Init_locks_ctor
@@ -328,9 +328,9 @@
 @ cdecl -arch=arm ??0_Timevec@std@@QAA@PAX@Z(ptr ptr) _Timevec_ctor_timeptr
 @ thiscall -arch=i386 ??0_Timevec@std@@QAE@PAX@Z(ptr ptr) _Timevec_ctor_timeptr
 @ cdecl -arch=win64 ??0_Timevec@std@@QEAA@PEAX@Z(ptr ptr) _Timevec_ctor_timeptr
-@ stub -arch=arm ??0_UShinit@std@@QAA@XZ
-@ stub -arch=i386 ??0_UShinit@std@@QAE@XZ
-@ stub -arch=win64 ??0_UShinit@std@@QEAA@XZ
+@ cdecl -arch=arm ??0_UShinit@std@@QAA@XZ(ptr) _UShinit_ctor
+@ thiscall -arch=i386 ??0_UShinit@std@@QAE@XZ(ptr) _UShinit_ctor
+@ cdecl -arch=win64 ??0_UShinit@std@@QEAA@XZ(ptr) _UShinit_ctor
 @ cdecl -arch=arm ??0_Winit@std@@QAA@XZ(ptr) _Winit_ctor
 @ thiscall -arch=i386 ??0_Winit@std@@QAE@XZ(ptr) _Winit_ctor
 @ cdecl -arch=win64 ??0_Winit@std@@QEAA@XZ(ptr) _Winit_ctor
@@ -481,9 +481,9 @@
 @ cdecl -arch=arm ??1_Container_base12@std@@QAA@XZ(ptr) _Container_base12_dtor
 @ thiscall -arch=i386 ??1_Container_base12@std@@QAE@XZ(ptr) _Container_base12_dtor
 @ cdecl -arch=win64 ??1_Container_base12@std@@QEAA@XZ(ptr) _Container_base12_dtor
-@ stub -arch=arm ??1_Facet_base@std@@UAA@XZ
-@ stub -arch=i386 ??1_Facet_base@std@@UAE@XZ
-@ stub -arch=win64 ??1_Facet_base@std@@UEAA@XZ
+@ cdecl -arch=arm ??1_Facet_base@std@@UAA@XZ(ptr) _Facet_base_dtor
+@ thiscall -arch=i386 ??1_Facet_base@std@@UAE@XZ(ptr) _Facet_base_dtor
+@ cdecl -arch=win64 ??1_Facet_base@std@@UEAA@XZ(ptr) _Facet_base_dtor
 @ cdecl -arch=arm ??1_Init_locks@std@@QAA@XZ(ptr) _Init_locks_dtor
 @ thiscall -arch=i386 ??1_Init_locks@std@@QAE@XZ(ptr) _Init_locks_dtor
 @ cdecl -arch=win64 ??1_Init_locks@std@@QEAA@XZ(ptr) _Init_locks_dtor
@@ -502,9 +502,9 @@
 @ cdecl -arch=arm ??1_Timevec@std@@QAA@XZ(ptr) _Timevec_dtor
 @ thiscall -arch=i386 ??1_Timevec@std@@QAE@XZ(ptr) _Timevec_dtor
 @ cdecl -arch=win64 ??1_Timevec@std@@QEAA@XZ(ptr) _Timevec_dtor
-@ stub -arch=arm ??1_UShinit@std@@QAA@XZ
-@ stub -arch=i386 ??1_UShinit@std@@QAE@XZ
-@ stub -arch=win64 ??1_UShinit@std@@QEAA@XZ
+@ cdecl -arch=arm ??1_UShinit@std@@QAA@XZ(ptr) _UShinit_dtor
+@ thiscall -arch=i386 ??1_UShinit@std@@QAE@XZ(ptr) _UShinit_dtor
+@ cdecl -arch=win64 ??1_UShinit@std@@QEAA@XZ(ptr) _UShinit_dtor
 @ cdecl -arch=arm ??1_Winit@std@@QAA@XZ(ptr) _Winit_dtor
 @ thiscall -arch=i386 ??1_Winit@std@@QAE@XZ(ptr) _Winit_dtor
 @ cdecl -arch=win64 ??1_Winit@std@@QEAA@XZ(ptr) _Winit_dtor
@@ -526,9 +526,9 @@
 @ cdecl -arch=arm ??1time_base@std@@UAA@XZ(ptr) time_base_dtor
 @ thiscall -arch=i386 ??1time_base@std@@UAE@XZ(ptr) time_base_dtor
 @ cdecl -arch=win64 ??1time_base@std@@UEAA@XZ(ptr) time_base_dtor
-@ stub -arch=arm ??4?$_Iosb@H@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Iosb@H@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Iosb@H@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4?$_Iosb@H@std@@QAAAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ thiscall -arch=i386 ??4?$_Iosb@H@std@@QAEAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=win64 ??4?$_Iosb@H@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) empty_class_op_assign
 @ cdecl -arch=arm ??4?$_Yarn@D@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Yarn_char_op_assign
 @ thiscall -arch=i386 ??4?$_Yarn@D@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Yarn_char_op_assign
 @ cdecl -arch=win64 ??4?$_Yarn@D@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Yarn_char_op_assign
@@ -583,21 +583,21 @@
 @ cdecl -arch=arm ??4_Container_base12@std@@QAAAAU01@ABU01@@Z(ptr ptr) _Container_base12_op_assign
 @ thiscall -arch=i386 ??4_Container_base12@std@@QAEAAU01@ABU01@@Z(ptr ptr) _Container_base12_op_assign
 @ cdecl -arch=win64 ??4_Container_base12@std@@QEAAAEAU01@AEBU01@@Z(ptr ptr) _Container_base12_op_assign
-@ stub -arch=arm ??4_Facet_base@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_Facet_base@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_Facet_base@std@@QEAAAEAV01@AEBV01@@Z
-@ stub -arch=arm ??4_Init_locks@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_Init_locks@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_Init_locks@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4_Facet_base@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Facet_base_op_assign
+@ thiscall -arch=i386 ??4_Facet_base@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Facet_base_op_assign
+@ cdecl -arch=win64 ??4_Facet_base@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Facet_base_op_assign
+@ cdecl -arch=arm ??4_Init_locks@std@@QAAAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ thiscall -arch=i386 ??4_Init_locks@std@@QAEAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=win64 ??4_Init_locks@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) empty_class_op_assign
 @ cdecl -arch=arm ??4_Pad@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Pad_op_assign
 @ thiscall -arch=i386 ??4_Pad@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Pad_op_assign
 @ cdecl -arch=win64 ??4_Pad@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Pad_op_assign
 @ cdecl -arch=arm ??4_Timevec@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Timevec_op_assign
 @ thiscall -arch=i386 ??4_Timevec@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Timevec_op_assign
 @ cdecl -arch=win64 ??4_Timevec@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Timevec_op_assign
-@ stub -arch=arm ??4_UShinit@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_UShinit@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_UShinit@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4_UShinit@std@@QAAAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ thiscall -arch=i386 ??4_UShinit@std@@QAEAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=win64 ??4_UShinit@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) empty_class_op_assign
 @ cdecl -arch=arm ??4_Winit@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Winit_op_assign
 @ thiscall -arch=i386 ??4_Winit@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Winit_op_assign
 @ cdecl -arch=win64 ??4_Winit@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Winit_op_assign
diff --git a/dlls/msvcp120/msvcp120.spec b/dlls/msvcp120/msvcp120.spec
index c6639e6..b1f121f 100644
--- a/dlls/msvcp120/msvcp120.spec
+++ b/dlls/msvcp120/msvcp120.spec
@@ -283,12 +283,12 @@
 @ cdecl -arch=arm ??0_Container_base12@std@@QAA@XZ(ptr) _Container_base12_ctor
 @ thiscall -arch=i386 ??0_Container_base12@std@@QAE@XZ(ptr) _Container_base12_ctor
 @ cdecl -arch=win64 ??0_Container_base12@std@@QEAA@XZ(ptr) _Container_base12_ctor
-@ stub -arch=arm ??0_Facet_base@std@@QAA@ABV01@@Z
-@ stub -arch=i386 ??0_Facet_base@std@@QAE@ABV01@@Z
-@ stub -arch=win64 ??0_Facet_base@std@@QEAA@AEBV01@@Z
-@ stub -arch=arm ??0_Facet_base@std@@QAA@XZ
-@ stub -arch=i386 ??0_Facet_base@std@@QAE@XZ
-@ stub -arch=win64 ??0_Facet_base@std@@QEAA@XZ
+@ cdecl -arch=arm ??0_Facet_base@std@@QAA@ABV01@@Z(ptr ptr) _Facet_base_ctor
+@ thiscall -arch=i386 ??0_Facet_base@std@@QAE@ABV01@@Z(ptr ptr) _Facet_base_ctor
+@ cdecl -arch=win64 ??0_Facet_base@std@@QEAA@AEBV01@@Z(ptr ptr) _Facet_base_ctor
+@ cdecl -arch=arm ??0_Facet_base@std@@QAA@XZ(ptr ptr) _Facet_base_ctor
+@ thiscall -arch=i386 ??0_Facet_base@std@@QAE@XZ(ptr ptr) _Facet_base_ctor
+@ cdecl -arch=win64 ??0_Facet_base@std@@QEAA@XZ(ptr ptr) _Facet_base_ctor
 @ cdecl -arch=arm ??0_Init_locks@std@@QAA@XZ(ptr) _Init_locks_ctor
 @ thiscall -arch=i386 ??0_Init_locks@std@@QAE@XZ(ptr) _Init_locks_ctor
 @ cdecl -arch=win64 ??0_Init_locks@std@@QEAA@XZ(ptr) _Init_locks_ctor
@@ -328,9 +328,9 @@
 @ cdecl -arch=arm ??0_Timevec@std@@QAA@PAX@Z(ptr ptr) _Timevec_ctor_timeptr
 @ thiscall -arch=i386 ??0_Timevec@std@@QAE@PAX@Z(ptr ptr) _Timevec_ctor_timeptr
 @ cdecl -arch=win64 ??0_Timevec@std@@QEAA@PEAX@Z(ptr ptr) _Timevec_ctor_timeptr
-@ stub -arch=arm ??0_UShinit@std@@QAA@XZ
-@ stub -arch=i386 ??0_UShinit@std@@QAE@XZ
-@ stub -arch=win64 ??0_UShinit@std@@QEAA@XZ
+@ cdecl -arch=arm ??0_UShinit@std@@QAA@XZ(ptr) _UShinit_ctor
+@ thiscall -arch=i386 ??0_UShinit@std@@QAE@XZ(ptr) _UShinit_ctor
+@ cdecl -arch=win64 ??0_UShinit@std@@QEAA@XZ(ptr) _UShinit_ctor
 @ cdecl -arch=arm ??0_Winit@std@@QAA@XZ(ptr) _Winit_ctor
 @ thiscall -arch=i386 ??0_Winit@std@@QAE@XZ(ptr) _Winit_ctor
 @ cdecl -arch=win64 ??0_Winit@std@@QEAA@XZ(ptr) _Winit_ctor
@@ -481,9 +481,9 @@
 @ cdecl -arch=arm ??1_Container_base12@std@@QAA@XZ(ptr) _Container_base12_dtor
 @ thiscall -arch=i386 ??1_Container_base12@std@@QAE@XZ(ptr) _Container_base12_dtor
 @ cdecl -arch=win64 ??1_Container_base12@std@@QEAA@XZ(ptr) _Container_base12_dtor
-@ stub -arch=arm ??1_Facet_base@std@@UAA@XZ
-@ stub -arch=i386 ??1_Facet_base@std@@UAE@XZ
-@ stub -arch=win64 ??1_Facet_base@std@@UEAA@XZ
+@ cdecl -arch=arm ??1_Facet_base@std@@UAA@XZ(ptr) _Facet_base_dtor
+@ thiscall -arch=i386 ??1_Facet_base@std@@UAE@XZ(ptr) _Facet_base_dtor
+@ cdecl -arch=win64 ??1_Facet_base@std@@UEAA@XZ(ptr) _Facet_base_dtor
 @ cdecl -arch=arm ??1_Init_locks@std@@QAA@XZ(ptr) _Init_locks_dtor
 @ thiscall -arch=i386 ??1_Init_locks@std@@QAE@XZ(ptr) _Init_locks_dtor
 @ cdecl -arch=win64 ??1_Init_locks@std@@QEAA@XZ(ptr) _Init_locks_dtor
@@ -502,9 +502,9 @@
 @ cdecl -arch=arm ??1_Timevec@std@@QAA@XZ(ptr) _Timevec_dtor
 @ thiscall -arch=i386 ??1_Timevec@std@@QAE@XZ(ptr) _Timevec_dtor
 @ cdecl -arch=win64 ??1_Timevec@std@@QEAA@XZ(ptr) _Timevec_dtor
-@ stub -arch=arm ??1_UShinit@std@@QAA@XZ
-@ stub -arch=i386 ??1_UShinit@std@@QAE@XZ
-@ stub -arch=win64 ??1_UShinit@std@@QEAA@XZ
+@ cdecl -arch=arm ??1_UShinit@std@@QAA@XZ(ptr) _UShinit_dtor
+@ thiscall -arch=i386 ??1_UShinit@std@@QAE@XZ(ptr) _UShinit_dtor
+@ cdecl -arch=win64 ??1_UShinit@std@@QEAA@XZ(ptr) _UShinit_dtor
 @ cdecl -arch=arm ??1_Winit@std@@QAA@XZ(ptr) _Winit_dtor
 @ thiscall -arch=i386 ??1_Winit@std@@QAE@XZ(ptr) _Winit_dtor
 @ cdecl -arch=win64 ??1_Winit@std@@QEAA@XZ(ptr) _Winit_dtor
@@ -526,9 +526,9 @@
 @ cdecl -arch=arm ??1time_base@std@@UAA@XZ(ptr) time_base_dtor
 @ thiscall -arch=i386 ??1time_base@std@@UAE@XZ(ptr) time_base_dtor
 @ cdecl -arch=win64 ??1time_base@std@@UEAA@XZ(ptr) time_base_dtor
-@ stub -arch=arm ??4?$_Iosb@H@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Iosb@H@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Iosb@H@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4?$_Iosb@H@std@@QAAAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ thiscall -arch=i386 ??4?$_Iosb@H@std@@QAEAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=win64 ??4?$_Iosb@H@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) empty_class_op_assign
 @ cdecl -arch=arm ??4?$_Yarn@D@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Yarn_char_op_assign
 @ thiscall -arch=i386 ??4?$_Yarn@D@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Yarn_char_op_assign
 @ cdecl -arch=win64 ??4?$_Yarn@D@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Yarn_char_op_assign
@@ -583,21 +583,21 @@
 @ cdecl -arch=arm ??4_Container_base12@std@@QAAAAU01@ABU01@@Z(ptr ptr) _Container_base12_op_assign
 @ thiscall -arch=i386 ??4_Container_base12@std@@QAEAAU01@ABU01@@Z(ptr ptr) _Container_base12_op_assign
 @ cdecl -arch=win64 ??4_Container_base12@std@@QEAAAEAU01@AEBU01@@Z(ptr ptr) _Container_base12_op_assign
-@ stub -arch=arm ??4_Facet_base@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_Facet_base@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_Facet_base@std@@QEAAAEAV01@AEBV01@@Z
-@ stub -arch=arm ??4_Init_locks@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_Init_locks@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_Init_locks@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4_Facet_base@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Facet_base_op_assign
+@ thiscall -arch=i386 ??4_Facet_base@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Facet_base_op_assign
+@ cdecl -arch=win64 ??4_Facet_base@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Facet_base_op_assign
+@ cdecl -arch=arm ??4_Init_locks@std@@QAAAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ thiscall -arch=i386 ??4_Init_locks@std@@QAEAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=win64 ??4_Init_locks@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) empty_class_op_assign
 @ cdecl -arch=arm ??4_Pad@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Pad_op_assign
 @ thiscall -arch=i386 ??4_Pad@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Pad_op_assign
 @ cdecl -arch=win64 ??4_Pad@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Pad_op_assign
 @ cdecl -arch=arm ??4_Timevec@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Timevec_op_assign
 @ thiscall -arch=i386 ??4_Timevec@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Timevec_op_assign
 @ cdecl -arch=win64 ??4_Timevec@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Timevec_op_assign
-@ stub -arch=arm ??4_UShinit@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_UShinit@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_UShinit@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4_UShinit@std@@QAAAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ thiscall -arch=i386 ??4_UShinit@std@@QAEAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=win64 ??4_UShinit@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) empty_class_op_assign
 @ cdecl -arch=arm ??4_Winit@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Winit_op_assign
 @ thiscall -arch=i386 ??4_Winit@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Winit_op_assign
 @ cdecl -arch=win64 ??4_Winit@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Winit_op_assign
diff --git a/dlls/msvcp120_app/msvcp120_app.spec b/dlls/msvcp120_app/msvcp120_app.spec
index e8f5641..314cf6a 100644
--- a/dlls/msvcp120_app/msvcp120_app.spec
+++ b/dlls/msvcp120_app/msvcp120_app.spec
@@ -283,12 +283,12 @@
 @ cdecl -arch=arm ??0_Container_base12@std@@QAA@XZ(ptr) msvcp120.??0_Container_base12@std@@QAA@XZ
 @ thiscall -arch=i386 ??0_Container_base12@std@@QAE@XZ(ptr) msvcp120.??0_Container_base12@std@@QAE@XZ
 @ cdecl -arch=win64 ??0_Container_base12@std@@QEAA@XZ(ptr) msvcp120.??0_Container_base12@std@@QEAA@XZ
-@ stub -arch=arm ??0_Facet_base@std@@QAA@ABV01@@Z
-@ stub -arch=i386 ??0_Facet_base@std@@QAE@ABV01@@Z
-@ stub -arch=win64 ??0_Facet_base@std@@QEAA@AEBV01@@Z
-@ stub -arch=arm ??0_Facet_base@std@@QAA@XZ
-@ stub -arch=i386 ??0_Facet_base@std@@QAE@XZ
-@ stub -arch=win64 ??0_Facet_base@std@@QEAA@XZ
+@ cdecl -arch=arm ??0_Facet_base@std@@QAA@ABV01@@Z(ptr ptr) msvcp120.??0_Facet_base@std@@QAA@ABV01@@Z
+@ thiscall -arch=i386 ??0_Facet_base@std@@QAE@ABV01@@Z(ptr ptr) msvcp120.??0_Facet_base@std@@QAE@ABV01@@Z
+@ cdecl -arch=win64 ??0_Facet_base@std@@QEAA@AEBV01@@Z(ptr ptr) msvcp120.??0_Facet_base@std@@QEAA@AEBV01@@Z
+@ cdecl -arch=arm ??0_Facet_base@std@@QAA@XZ(ptr ptr) msvcp120.??0_Facet_base@std@@QAA@XZ
+@ thiscall -arch=i386 ??0_Facet_base@std@@QAE@XZ(ptr ptr) msvcp120.??0_Facet_base@std@@QAE@XZ
+@ cdecl -arch=win64 ??0_Facet_base@std@@QEAA@XZ(ptr ptr) msvcp120.??0_Facet_base@std@@QEAA@XZ
 @ cdecl -arch=arm ??0_Init_locks@std@@QAA@XZ(ptr) msvcp120.??0_Init_locks@std@@QAA@XZ
 @ thiscall -arch=i386 ??0_Init_locks@std@@QAE@XZ(ptr) msvcp120.??0_Init_locks@std@@QAE@XZ
 @ cdecl -arch=win64 ??0_Init_locks@std@@QEAA@XZ(ptr) msvcp120.??0_Init_locks@std@@QEAA@XZ
@@ -328,9 +328,9 @@
 @ cdecl -arch=arm ??0_Timevec@std@@QAA@PAX@Z(ptr ptr) msvcp120.??0_Timevec@std@@QAA@PAX@Z
 @ thiscall -arch=i386 ??0_Timevec@std@@QAE@PAX@Z(ptr ptr) msvcp120.??0_Timevec@std@@QAE@PAX@Z
 @ cdecl -arch=win64 ??0_Timevec@std@@QEAA@PEAX@Z(ptr ptr) msvcp120.??0_Timevec@std@@QEAA@PEAX@Z
-@ stub -arch=arm ??0_UShinit@std@@QAA@XZ
-@ stub -arch=i386 ??0_UShinit@std@@QAE@XZ
-@ stub -arch=win64 ??0_UShinit@std@@QEAA@XZ
+@ cdecl -arch=arm ??0_UShinit@std@@QAA@XZ(ptr) msvcp120.??0_UShinit@std@@QAA@XZ
+@ thiscall -arch=i386 ??0_UShinit@std@@QAE@XZ(ptr) msvcp120.??0_UShinit@std@@QAE@XZ
+@ cdecl -arch=win64 ??0_UShinit@std@@QEAA@XZ(ptr) msvcp120.??0_UShinit@std@@QEAA@XZ
 @ cdecl -arch=arm ??0_Winit@std@@QAA@XZ(ptr) msvcp120.??0_Winit@std@@QAA@XZ
 @ thiscall -arch=i386 ??0_Winit@std@@QAE@XZ(ptr) msvcp120.??0_Winit@std@@QAE@XZ
 @ cdecl -arch=win64 ??0_Winit@std@@QEAA@XZ(ptr) msvcp120.??0_Winit@std@@QEAA@XZ
@@ -481,9 +481,9 @@
 @ cdecl -arch=arm ??1_Container_base12@std@@QAA@XZ(ptr) msvcp120.??1_Container_base12@std@@QAA@XZ
 @ thiscall -arch=i386 ??1_Container_base12@std@@QAE@XZ(ptr) msvcp120.??1_Container_base12@std@@QAE@XZ
 @ cdecl -arch=win64 ??1_Container_base12@std@@QEAA@XZ(ptr) msvcp120.??1_Container_base12@std@@QEAA@XZ
-@ stub -arch=arm ??1_Facet_base@std@@UAA@XZ
-@ stub -arch=i386 ??1_Facet_base@std@@UAE@XZ
-@ stub -arch=win64 ??1_Facet_base@std@@UEAA@XZ
+@ cdecl -arch=arm ??1_Facet_base@std@@UAA@XZ(ptr) msvcp120.??1_Facet_base@std@@UAA@XZ
+@ thiscall -arch=i386 ??1_Facet_base@std@@UAE@XZ(ptr) msvcp120.??1_Facet_base@std@@UAE@XZ
+@ cdecl -arch=win64 ??1_Facet_base@std@@UEAA@XZ(ptr) msvcp120.??1_Facet_base@std@@UEAA@XZ
 @ cdecl -arch=arm ??1_Init_locks@std@@QAA@XZ(ptr) msvcp120.??1_Init_locks@std@@QAA@XZ
 @ thiscall -arch=i386 ??1_Init_locks@std@@QAE@XZ(ptr) msvcp120.??1_Init_locks@std@@QAE@XZ
 @ cdecl -arch=win64 ??1_Init_locks@std@@QEAA@XZ(ptr) msvcp120.??1_Init_locks@std@@QEAA@XZ
@@ -502,9 +502,9 @@
 @ cdecl -arch=arm ??1_Timevec@std@@QAA@XZ(ptr) msvcp120.??1_Timevec@std@@QAA@XZ
 @ thiscall -arch=i386 ??1_Timevec@std@@QAE@XZ(ptr) msvcp120.??1_Timevec@std@@QAE@XZ
 @ cdecl -arch=win64 ??1_Timevec@std@@QEAA@XZ(ptr) msvcp120.??1_Timevec@std@@QEAA@XZ
-@ stub -arch=arm ??1_UShinit@std@@QAA@XZ
-@ stub -arch=i386 ??1_UShinit@std@@QAE@XZ
-@ stub -arch=win64 ??1_UShinit@std@@QEAA@XZ
+@ cdecl -arch=arm ??1_UShinit@std@@QAA@XZ(ptr) msvcp120.??1_UShinit@std@@QAA@XZ
+@ thiscall -arch=i386 ??1_UShinit@std@@QAE@XZ(ptr) msvcp120.??1_UShinit@std@@QAE@XZ
+@ cdecl -arch=win64 ??1_UShinit@std@@QEAA@XZ(ptr) msvcp120.??1_UShinit@std@@QEAA@XZ
 @ cdecl -arch=arm ??1_Winit@std@@QAA@XZ(ptr) msvcp120.??1_Winit@std@@QAA@XZ
 @ thiscall -arch=i386 ??1_Winit@std@@QAE@XZ(ptr) msvcp120.??1_Winit@std@@QAE@XZ
 @ cdecl -arch=win64 ??1_Winit@std@@QEAA@XZ(ptr) msvcp120.??1_Winit@std@@QEAA@XZ
@@ -526,9 +526,9 @@
 @ cdecl -arch=arm ??1time_base@std@@UAA@XZ(ptr) msvcp120.??1time_base@std@@UAA@XZ
 @ thiscall -arch=i386 ??1time_base@std@@UAE@XZ(ptr) msvcp120.??1time_base@std@@UAE@XZ
 @ cdecl -arch=win64 ??1time_base@std@@UEAA@XZ(ptr) msvcp120.??1time_base@std@@UEAA@XZ
-@ stub -arch=arm ??4?$_Iosb@H@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Iosb@H@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Iosb@H@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4?$_Iosb@H@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp120.??4?$_Iosb@H@std@@QAAAAV01@ABV01@@Z
+@ thiscall -arch=i386 ??4?$_Iosb@H@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp120.??4?$_Iosb@H@std@@QAEAAV01@ABV01@@Z
+@ cdecl -arch=win64 ??4?$_Iosb@H@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp120.??4?$_Iosb@H@std@@QEAAAEAV01@AEBV01@@Z
 @ cdecl -arch=arm ??4?$_Yarn@D@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp120.??4?$_Yarn@D@std@@QAAAAV01@ABV01@@Z
 @ thiscall -arch=i386 ??4?$_Yarn@D@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp120.??4?$_Yarn@D@std@@QAEAAV01@ABV01@@Z
 @ cdecl -arch=win64 ??4?$_Yarn@D@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp120.??4?$_Yarn@D@std@@QEAAAEAV01@AEBV01@@Z
@@ -583,21 +583,21 @@
 @ cdecl -arch=arm ??4_Container_base12@std@@QAAAAU01@ABU01@@Z(ptr ptr) msvcp120.??4_Container_base12@std@@QAAAAU01@ABU01@@Z
 @ thiscall -arch=i386 ??4_Container_base12@std@@QAEAAU01@ABU01@@Z(ptr ptr) msvcp120.??4_Container_base12@std@@QAEAAU01@ABU01@@Z
 @ cdecl -arch=win64 ??4_Container_base12@std@@QEAAAEAU01@AEBU01@@Z(ptr ptr) msvcp120.??4_Container_base12@std@@QEAAAEAU01@AEBU01@@Z
-@ stub -arch=arm ??4_Facet_base@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_Facet_base@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_Facet_base@std@@QEAAAEAV01@AEBV01@@Z
-@ stub -arch=arm ??4_Init_locks@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_Init_locks@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_Init_locks@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4_Facet_base@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp120.??4_Facet_base@std@@QAAAAV01@ABV01@@Z
+@ thiscall -arch=i386 ??4_Facet_base@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp120.??4_Facet_base@std@@QAEAAV01@ABV01@@Z
+@ cdecl -arch=win64 ??4_Facet_base@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp120.??4_Facet_base@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4_Init_locks@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp120.??4_Init_locks@std@@QAAAAV01@ABV01@@Z
+@ thiscall -arch=i386 ??4_Init_locks@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp120.??4_Init_locks@std@@QAEAAV01@ABV01@@Z
+@ cdecl -arch=win64 ??4_Init_locks@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp120.??4_Init_locks@std@@QEAAAEAV01@AEBV01@@Z
 @ cdecl -arch=arm ??4_Pad@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp120.??4_Pad@std@@QAAAAV01@ABV01@@Z
 @ thiscall -arch=i386 ??4_Pad@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp120.??4_Pad@std@@QAEAAV01@ABV01@@Z
 @ cdecl -arch=win64 ??4_Pad@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp120.??4_Pad@std@@QEAAAEAV01@AEBV01@@Z
 @ cdecl -arch=arm ??4_Timevec@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp120.??4_Timevec@std@@QAAAAV01@ABV01@@Z
 @ thiscall -arch=i386 ??4_Timevec@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp120.??4_Timevec@std@@QAEAAV01@ABV01@@Z
 @ cdecl -arch=win64 ??4_Timevec@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp120.??4_Timevec@std@@QEAAAEAV01@AEBV01@@Z
-@ stub -arch=arm ??4_UShinit@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_UShinit@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_UShinit@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4_UShinit@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp120.??4_UShinit@std@@QAAAAV01@ABV01@@Z
+@ thiscall -arch=i386 ??4_UShinit@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp120.??4_UShinit@std@@QAEAAV01@ABV01@@Z
+@ cdecl -arch=win64 ??4_UShinit@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp120.??4_UShinit@std@@QEAAAEAV01@AEBV01@@Z
 @ cdecl -arch=arm ??4_Winit@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp120.??4_Winit@std@@QAAAAV01@ABV01@@Z
 @ thiscall -arch=i386 ??4_Winit@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp120.??4_Winit@std@@QAEAAV01@ABV01@@Z
 @ cdecl -arch=win64 ??4_Winit@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp120.??4_Winit@std@@QEAAAEAV01@AEBV01@@Z
diff --git a/dlls/msvcp140/msvcp140.spec b/dlls/msvcp140/msvcp140.spec
index 7f9f00a..91e23d1 100644
--- a/dlls/msvcp140/msvcp140.spec
+++ b/dlls/msvcp140/msvcp140.spec
@@ -16,21 +16,21 @@
 @ cdecl -arch=arm ??0?$_Yarn@D@std@@QAA@XZ(ptr) _Yarn_char_ctor
 @ thiscall -arch=i386 ??0?$_Yarn@D@std@@QAE@XZ(ptr) _Yarn_char_ctor
 @ cdecl -arch=win64 ??0?$_Yarn@D@std@@QEAA@XZ(ptr) _Yarn_char_ctor
-@ stub -arch=arm ??0?$_Yarn@G@std@@QAA@ABV01@@Z
-@ stub -arch=i386 ??0?$_Yarn@G@std@@QAE@ABV01@@Z
-@ stub -arch=win64 ??0?$_Yarn@G@std@@QEAA@AEBV01@@Z
-@ stub -arch=arm ??0?$_Yarn@G@std@@QAA@PBG@Z
-@ stub -arch=i386 ??0?$_Yarn@G@std@@QAE@PBG@Z
-@ stub -arch=win64 ??0?$_Yarn@G@std@@QEAA@PEBG@Z
-@ stub -arch=arm ??0?$_Yarn@G@std@@QAA@XZ
-@ stub -arch=i386 ??0?$_Yarn@G@std@@QAE@XZ
-@ stub -arch=win64 ??0?$_Yarn@G@std@@QEAA@XZ
-@ stub -arch=arm ??0?$_Yarn@_W@std@@QAA@ABV01@@Z
-@ stub -arch=i386 ??0?$_Yarn@_W@std@@QAE@ABV01@@Z
-@ stub -arch=win64 ??0?$_Yarn@_W@std@@QEAA@AEBV01@@Z
-@ stub -arch=arm ??0?$_Yarn@_W@std@@QAA@PB_W@Z
-@ stub -arch=i386 ??0?$_Yarn@_W@std@@QAE@PB_W@Z
-@ stub -arch=win64 ??0?$_Yarn@_W@std@@QEAA@PEB_W@Z
+@ cdecl -arch=arm ??0?$_Yarn@G@std@@QAA@ABV01@@Z(ptr ptr) _Yarn_wchar_copy_ctor
+@ thiscall -arch=i386 ??0?$_Yarn@G@std@@QAE@ABV01@@Z(ptr ptr) _Yarn_wchar_copy_ctor
+@ cdecl -arch=win64 ??0?$_Yarn@G@std@@QEAA@AEBV01@@Z(ptr ptr) _Yarn_wchar_copy_ctor
+@ cdecl -arch=arm ??0?$_Yarn@G@std@@QAA@PBG@Z(ptr wstr) _Yarn_wchar_ctor_cstr
+@ thiscall -arch=i386 ??0?$_Yarn@G@std@@QAE@PBG@Z(ptr wstr) _Yarn_wchar_ctor_cstr
+@ cdecl -arch=win64 ??0?$_Yarn@G@std@@QEAA@PEBG@Z(ptr wstr) _Yarn_wchar_ctor_cstr
+@ cdecl -arch=arm ??0?$_Yarn@G@std@@QAA@XZ(ptr) _Yarn_wchar_ctor
+@ thiscall -arch=i386 ??0?$_Yarn@G@std@@QAE@XZ(ptr) _Yarn_wchar_ctor
+@ cdecl -arch=win64 ??0?$_Yarn@G@std@@QEAA@XZ(ptr) _Yarn_wchar_ctor
+@ cdecl -arch=arm ??0?$_Yarn@_W@std@@QAA@ABV01@@Z(ptr ptr) _Yarn_wchar_copy_ctor
+@ thiscall -arch=i386 ??0?$_Yarn@_W@std@@QAE@ABV01@@Z(ptr ptr) _Yarn_wchar_copy_ctor
+@ cdecl -arch=win64 ??0?$_Yarn@_W@std@@QEAA@AEBV01@@Z(ptr ptr) _Yarn_wchar_copy_ctor
+@ cdecl -arch=arm ??0?$_Yarn@_W@std@@QAA@PB_W@Z(ptr wstr) _Yarn_wchar_ctor_cstr
+@ thiscall -arch=i386 ??0?$_Yarn@_W@std@@QAE@PB_W@Z(ptr wstr) _Yarn_wchar_ctor_cstr
+@ cdecl -arch=win64 ??0?$_Yarn@_W@std@@QEAA@PEB_W@Z(ptr wstr) _Yarn_wchar_ctor_cstr
 @ cdecl -arch=arm ??0?$_Yarn@_W@std@@QAA@XZ(ptr) _Yarn_wchar_ctor
 @ thiscall -arch=i386 ??0?$_Yarn@_W@std@@QAE@XZ(ptr) _Yarn_wchar_ctor
 @ cdecl -arch=win64 ??0?$_Yarn@_W@std@@QEAA@XZ(ptr) _Yarn_wchar_ctor
@@ -304,12 +304,12 @@
 @ cdecl -arch=arm ??0Init@ios_base@std@@QAA@XZ(ptr) ios_base_Init_ctor
 @ thiscall -arch=i386 ??0Init@ios_base@std@@QAE@XZ(ptr) ios_base_Init_ctor
 @ cdecl -arch=win64 ??0Init@ios_base@std@@QEAA@XZ(ptr) ios_base_Init_ctor
-@ stub -arch=arm ??0_Facet_base@std@@QAA@ABV01@@Z
-@ stub -arch=i386 ??0_Facet_base@std@@QAE@ABV01@@Z
-@ stub -arch=win64 ??0_Facet_base@std@@QEAA@AEBV01@@Z
-@ stub -arch=arm ??0_Facet_base@std@@QAA@XZ
-@ stub -arch=i386 ??0_Facet_base@std@@QAE@XZ
-@ stub -arch=win64 ??0_Facet_base@std@@QEAA@XZ
+@ cdecl -arch=arm ??0_Facet_base@std@@QAA@ABV01@@Z(ptr ptr) _Facet_base_ctor
+@ thiscall -arch=i386 ??0_Facet_base@std@@QAE@ABV01@@Z(ptr ptr) _Facet_base_ctor
+@ cdecl -arch=win64 ??0_Facet_base@std@@QEAA@AEBV01@@Z(ptr ptr) _Facet_base_ctor
+@ cdecl -arch=arm ??0_Facet_base@std@@QAA@XZ(ptr ptr) _Facet_base_ctor
+@ thiscall -arch=i386 ??0_Facet_base@std@@QAE@XZ(ptr ptr) _Facet_base_ctor
+@ cdecl -arch=win64 ??0_Facet_base@std@@QEAA@XZ(ptr ptr) _Facet_base_ctor
 @ cdecl -arch=arm ??0_Init_locks@std@@QAA@XZ(ptr) _Init_locks_ctor
 @ thiscall -arch=i386 ??0_Init_locks@std@@QAE@XZ(ptr) _Init_locks_ctor
 @ cdecl -arch=win64 ??0_Init_locks@std@@QEAA@XZ(ptr) _Init_locks_ctor
@@ -337,9 +337,9 @@
 @ cdecl -arch=arm ??0_Timevec@std@@QAA@PAX@Z(ptr ptr) _Timevec_ctor_timeptr
 @ thiscall -arch=i386 ??0_Timevec@std@@QAE@PAX@Z(ptr ptr) _Timevec_ctor_timeptr
 @ cdecl -arch=win64 ??0_Timevec@std@@QEAA@PEAX@Z(ptr ptr) _Timevec_ctor_timeptr
-@ stub -arch=arm ??0_UShinit@std@@QAA@XZ
-@ stub -arch=i386 ??0_UShinit@std@@QAE@XZ
-@ stub -arch=win64 ??0_UShinit@std@@QEAA@XZ
+@ cdecl -arch=arm ??0_UShinit@std@@QAA@XZ(ptr) _UShinit_ctor
+@ thiscall -arch=i386 ??0_UShinit@std@@QAE@XZ(ptr) _UShinit_ctor
+@ cdecl -arch=win64 ??0_UShinit@std@@QEAA@XZ(ptr) _UShinit_ctor
 @ cdecl -arch=arm ??0_Winit@std@@QAA@XZ(ptr) _Winit_ctor
 @ thiscall -arch=i386 ??0_Winit@std@@QAE@XZ(ptr) _Winit_ctor
 @ cdecl -arch=win64 ??0_Winit@std@@QEAA@XZ(ptr) _Winit_ctor
@@ -367,9 +367,9 @@
 @ cdecl -arch=arm ??1?$_Yarn@D@std@@QAA@XZ(ptr) _Yarn_char_dtor
 @ thiscall -arch=i386 ??1?$_Yarn@D@std@@QAE@XZ(ptr) _Yarn_char_dtor
 @ cdecl -arch=win64 ??1?$_Yarn@D@std@@QEAA@XZ(ptr) _Yarn_char_dtor
-@ stub -arch=arm ??1?$_Yarn@G@std@@QAA@XZ
-@ stub -arch=i386 ??1?$_Yarn@G@std@@QAE@XZ
-@ stub -arch=win64 ??1?$_Yarn@G@std@@QEAA@XZ
+@ cdecl -arch=arm ??1?$_Yarn@G@std@@QAA@XZ(ptr) _Yarn_wchar_dtor
+@ thiscall -arch=i386 ??1?$_Yarn@G@std@@QAE@XZ(ptr) _Yarn_wchar_dtor
+@ cdecl -arch=win64 ??1?$_Yarn@G@std@@QEAA@XZ(ptr) _Yarn_wchar_dtor
 @ cdecl -arch=arm ??1?$_Yarn@_W@std@@QAA@XZ(ptr) _Yarn_wchar_dtor
 @ thiscall -arch=i386 ??1?$_Yarn@_W@std@@QAE@XZ(ptr) _Yarn_wchar_dtor
 @ cdecl -arch=win64 ??1?$_Yarn@_W@std@@QEAA@XZ(ptr) _Yarn_wchar_dtor
@@ -481,9 +481,9 @@
 @ cdecl -arch=arm ??1Init@ios_base@std@@QAA@XZ(ptr) ios_base_Init_dtor
 @ thiscall -arch=i386 ??1Init@ios_base@std@@QAE@XZ(ptr) ios_base_Init_dtor
 @ cdecl -arch=win64 ??1Init@ios_base@std@@QEAA@XZ(ptr) ios_base_Init_dtor
-@ stub -arch=arm ??1_Facet_base@std@@UAA@XZ
-@ stub -arch=i386 ??1_Facet_base@std@@UAE@XZ
-@ stub -arch=win64 ??1_Facet_base@std@@UEAA@XZ
+@ cdecl -arch=arm ??1_Facet_base@std@@UAA@XZ(ptr) _Facet_base_dtor
+@ thiscall -arch=i386 ??1_Facet_base@std@@UAE@XZ(ptr) _Facet_base_dtor
+@ cdecl -arch=win64 ??1_Facet_base@std@@UEAA@XZ(ptr) _Facet_base_dtor
 @ cdecl -arch=arm ??1_Init_locks@std@@QAA@XZ(ptr) _Init_locks_dtor
 @ thiscall -arch=i386 ??1_Init_locks@std@@QAE@XZ(ptr) _Init_locks_dtor
 @ cdecl -arch=win64 ??1_Init_locks@std@@QEAA@XZ(ptr) _Init_locks_dtor
@@ -499,9 +499,9 @@
 @ cdecl -arch=arm ??1_Timevec@std@@QAA@XZ(ptr) _Timevec_dtor
 @ thiscall -arch=i386 ??1_Timevec@std@@QAE@XZ(ptr) _Timevec_dtor
 @ cdecl -arch=win64 ??1_Timevec@std@@QEAA@XZ(ptr) _Timevec_dtor
-@ stub -arch=arm ??1_UShinit@std@@QAA@XZ
-@ stub -arch=i386 ??1_UShinit@std@@QAE@XZ
-@ stub -arch=win64 ??1_UShinit@std@@QEAA@XZ
+@ cdecl -arch=arm ??1_UShinit@std@@QAA@XZ(ptr) _UShinit_dtor
+@ thiscall -arch=i386 ??1_UShinit@std@@QAE@XZ(ptr) _UShinit_dtor
+@ cdecl -arch=win64 ??1_UShinit@std@@QEAA@XZ(ptr) _UShinit_dtor
 @ cdecl -arch=arm ??1_Winit@std@@QAA@XZ(ptr) _Winit_dtor
 @ thiscall -arch=i386 ??1_Winit@std@@QAE@XZ(ptr) _Winit_dtor
 @ cdecl -arch=win64 ??1_Winit@std@@QEAA@XZ(ptr) _Winit_dtor
@@ -520,28 +520,28 @@
 @ cdecl -arch=arm ??1time_base@std@@UAA@XZ(ptr) time_base_dtor
 @ thiscall -arch=i386 ??1time_base@std@@UAE@XZ(ptr) time_base_dtor
 @ cdecl -arch=win64 ??1time_base@std@@UEAA@XZ(ptr) time_base_dtor
-@ stub -arch=arm ??4?$_Iosb@H@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Iosb@H@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=arm ??4?$_Iosb@H@std@@QAAAAV01@$$QAV01@@Z
-@ stub -arch=i386 ??4?$_Iosb@H@std@@QAEAAV01@$$QAV01@@Z
-@ stub -arch=win64 ??4?$_Iosb@H@std@@QAEAAV01@$$QAV01@@Z
-@ stub -arch=arm ??4?$_Iosb@H@std@@QEAAAEAV01@$$QEAV01@@Z
-@ stub -arch=win64 ??4?$_Iosb@H@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4?$_Iosb@H@std@@QAAAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ thiscall -arch=i386 ??4?$_Iosb@H@std@@QAEAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=arm ??4?$_Iosb@H@std@@QAAAAV01@$$QAV01@@Z(ptr ptr) empty_class_op_assign
+@ thiscall -arch=i386 ??4?$_Iosb@H@std@@QAEAAV01@$$QAV01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=win64 ??4?$_Iosb@H@std@@QAEAAV01@$$QAV01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=arm ??4?$_Iosb@H@std@@QEAAAEAV01@$$QEAV01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=win64 ??4?$_Iosb@H@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) empty_class_op_assign
 @ cdecl -arch=arm ??4?$_Yarn@D@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Yarn_char_op_assign
 @ thiscall -arch=i386 ??4?$_Yarn@D@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Yarn_char_op_assign
 @ cdecl -arch=win64 ??4?$_Yarn@D@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Yarn_char_op_assign
 @ cdecl -arch=arm ??4?$_Yarn@D@std@@QAAAAV01@PBD@Z(ptr str) _Yarn_char_op_assign_cstr
 @ thiscall -arch=i386 ??4?$_Yarn@D@std@@QAEAAV01@PBD@Z(ptr str) _Yarn_char_op_assign_cstr
 @ cdecl -arch=win64 ??4?$_Yarn@D@std@@QEAAAEAV01@PEBD@Z(ptr str) _Yarn_char_op_assign_cstr
-@ stub -arch=arm ??4?$_Yarn@G@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Yarn@G@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Yarn@G@std@@QEAAAEAV01@AEBV01@@Z
-@ stub -arch=arm ??4?$_Yarn@G@std@@QAAAAV01@PBG@Z
-@ stub -arch=i386 ??4?$_Yarn@G@std@@QAEAAV01@PBG@Z
-@ stub -arch=win64 ??4?$_Yarn@G@std@@QEAAAEAV01@PEBG@Z
-@ stub -arch=arm ??4?$_Yarn@_W@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Yarn@_W@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Yarn@_W@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4?$_Yarn@G@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Yarn_wchar_op_assign
+@ thiscall -arch=i386 ??4?$_Yarn@G@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Yarn_wchar_op_assign
+@ cdecl -arch=win64 ??4?$_Yarn@G@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Yarn_wchar_op_assign
+@ cdecl -arch=arm ??4?$_Yarn@G@std@@QAAAAV01@PBG@Z(ptr wstr) _Yarn_wchar_op_assign_cstr
+@ thiscall -arch=i386 ??4?$_Yarn@G@std@@QAEAAV01@PBG@Z(ptr wstr) _Yarn_wchar_op_assign_cstr
+@ cdecl -arch=win64 ??4?$_Yarn@G@std@@QEAAAEAV01@PEBG@Z(ptr wstr) _Yarn_wchar_op_assign_cstr
+@ cdecl -arch=arm ??4?$_Yarn@_W@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Yarn_wchar_op_assign
+@ thiscall -arch=i386 ??4?$_Yarn@_W@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Yarn_wchar_op_assign
+@ cdecl -arch=win64 ??4?$_Yarn@_W@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Yarn_wchar_op_assign
 @ cdecl -arch=arm ??4?$_Yarn@_W@std@@QAAAAV01@PB_W@Z(ptr wstr) _Yarn_wchar_op_assign_cstr
 @ thiscall -arch=i386 ??4?$_Yarn@_W@std@@QAEAAV01@PB_W@Z(ptr wstr) _Yarn_wchar_op_assign_cstr
 @ cdecl -arch=win64 ??4?$_Yarn@_W@std@@QEAAAEAV01@PEB_W@Z(ptr wstr) _Yarn_wchar_op_assign_cstr
@@ -584,24 +584,24 @@
 @ cdecl -arch=arm ??4Init@ios_base@std@@QAAAAV012@ABV012@@Z(ptr ptr) ios_base_Init_op_assign
 @ thiscall -arch=i386 ??4Init@ios_base@std@@QAEAAV012@ABV012@@Z(ptr ptr) ios_base_Init_op_assign
 @ cdecl -arch=win64 ??4Init@ios_base@std@@QEAAAEAV012@AEBV012@@Z(ptr ptr) ios_base_Init_op_assign
-@ stub -arch=arm ??4_Crt_new_delete@std@@QAAAAU01@$$QAU01@@Z
-@ stub -arch=i386 ??4_Crt_new_delete@std@@QAEAAU01@$$QAU01@@Z
-@ stub -arch=win64 ??4_Crt_new_delete@std@@QEAAAEAU01@$$QEAU01@@Z
-@ stub -arch=arm ??4_Crt_new_delete@std@@QAAAAU01@ABU01@@Z
-@ stub -arch=i386 ??4_Crt_new_delete@std@@QAEAAU01@ABU01@@Z
-@ stub -arch=win64 ??4_Crt_new_delete@std@@QEAAAEAU01@AEBU01@@Z
-@ stub -arch=arm ??4_Facet_base@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_Facet_base@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_Facet_base@std@@QEAAAEAV01@AEBV01@@Z
-@ stub -arch=arm ??4_Init_locks@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_Init_locks@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_Init_locks@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4_Crt_new_delete@std@@QAAAAU01@$$QAU01@@Z(ptr ptr) empty_class_op_assign
+@ thiscall -arch=i386 ??4_Crt_new_delete@std@@QAEAAU01@$$QAU01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=win64 ??4_Crt_new_delete@std@@QEAAAEAU01@$$QEAU01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=arm ??4_Crt_new_delete@std@@QAAAAU01@ABU01@@Z(ptr ptr) empty_class_op_assign
+@ thiscall -arch=i386 ??4_Crt_new_delete@std@@QAEAAU01@ABU01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=win64 ??4_Crt_new_delete@std@@QEAAAEAU01@AEBU01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=arm ??4_Facet_base@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Facet_base_op_assign
+@ thiscall -arch=i386 ??4_Facet_base@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Facet_base_op_assign
+@ cdecl -arch=win64 ??4_Facet_base@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Facet_base_op_assign
+@ cdecl -arch=arm ??4_Init_locks@std@@QAAAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ thiscall -arch=i386 ??4_Init_locks@std@@QAEAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=win64 ??4_Init_locks@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) empty_class_op_assign
 @ cdecl -arch=arm ??4_Timevec@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Timevec_op_assign
 @ thiscall -arch=i386 ??4_Timevec@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Timevec_op_assign
 @ cdecl -arch=win64 ??4_Timevec@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Timevec_op_assign
-@ stub -arch=arm ??4_UShinit@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_UShinit@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_UShinit@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4_UShinit@std@@QAAAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ thiscall -arch=i386 ??4_UShinit@std@@QAEAAV01@ABV01@@Z(ptr ptr) empty_class_op_assign
+@ cdecl -arch=win64 ??4_UShinit@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) empty_class_op_assign
 @ cdecl -arch=arm ??4_Winit@std@@QAAAAV01@ABV01@@Z(ptr ptr) _Winit_op_assign
 @ thiscall -arch=i386 ??4_Winit@std@@QAEAAV01@ABV01@@Z(ptr ptr) _Winit_op_assign
 @ cdecl -arch=win64 ??4_Winit@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) _Winit_op_assign
@@ -1105,9 +1105,9 @@
 @ cdecl -arch=arm ?_C_str@?$_Yarn@D@std@@QBAPBDXZ(ptr) _Yarn_char_c_str
 @ thiscall -arch=i386 ?_C_str@?$_Yarn@D@std@@QBEPBDXZ(ptr) _Yarn_char_c_str
 @ cdecl -arch=win64 ?_C_str@?$_Yarn@D@std@@QEBAPEBDXZ(ptr) _Yarn_char_c_str
-@ stub -arch=arm ?_C_str@?$_Yarn@G@std@@QBAPBGXZ
-@ stub -arch=i386 ?_C_str@?$_Yarn@G@std@@QBEPBGXZ
-@ stub -arch=win64 ?_C_str@?$_Yarn@G@std@@QEBAPEBGXZ
+@ cdecl -arch=arm ?_C_str@?$_Yarn@G@std@@QBAPBGXZ(ptr) _Yarn_wchar__C_str
+@ thiscall -arch=i386 ?_C_str@?$_Yarn@G@std@@QBEPBGXZ(ptr) _Yarn_wchar__C_str
+@ cdecl -arch=win64 ?_C_str@?$_Yarn@G@std@@QEBAPEBGXZ(ptr) _Yarn_wchar__C_str
 @ cdecl -arch=arm ?_C_str@?$_Yarn@_W@std@@QBAPB_WXZ(ptr) _Yarn_wchar__C_str
 @ thiscall -arch=i386 ?_C_str@?$_Yarn@_W@std@@QBEPB_WXZ(ptr) _Yarn_wchar__C_str
 @ cdecl -arch=win64 ?_C_str@?$_Yarn@_W@std@@QEBAPEB_WXZ(ptr) _Yarn_wchar__C_str
@@ -1140,9 +1140,9 @@
 @ cdecl -arch=arm ?_Empty@?$_Yarn@D@std@@QBA_NXZ(ptr) _Yarn_char_empty
 @ thiscall -arch=i386 ?_Empty@?$_Yarn@D@std@@QBE_NXZ(ptr) _Yarn_char_empty
 @ cdecl -arch=win64 ?_Empty@?$_Yarn@D@std@@QEBA_NXZ(ptr) _Yarn_char_empty
-@ stub -arch=arm ?_Empty@?$_Yarn@G@std@@QBA_NXZ
-@ stub -arch=i386 ?_Empty@?$_Yarn@G@std@@QBE_NXZ
-@ stub -arch=win64 ?_Empty@?$_Yarn@G@std@@QEBA_NXZ
+@ cdecl -arch=arm ?_Empty@?$_Yarn@G@std@@QBA_NXZ(ptr) _Yarn_wchar__Empty
+@ thiscall -arch=i386 ?_Empty@?$_Yarn@G@std@@QBE_NXZ(ptr) _Yarn_wchar__Empty
+@ cdecl -arch=win64 ?_Empty@?$_Yarn@G@std@@QEBA_NXZ(ptr) _Yarn_wchar__Empty
 @ cdecl -arch=arm ?_Empty@?$_Yarn@_W@std@@QBA_NXZ(ptr) _Yarn_wchar__Empty
 @ thiscall -arch=i386 ?_Empty@?$_Yarn@_W@std@@QBE_NXZ(ptr) _Yarn_wchar__Empty
 @ cdecl -arch=win64 ?_Empty@?$_Yarn@_W@std@@QEBA_NXZ(ptr) _Yarn_wchar__Empty
@@ -1631,9 +1631,9 @@
 @ cdecl -arch=arm ?_Tidy@?$_Yarn@D@std@@AAAXXZ(ptr) _Yarn_char__Tidy
 @ thiscall -arch=i386 ?_Tidy@?$_Yarn@D@std@@AAEXXZ(ptr) _Yarn_char__Tidy
 @ cdecl -arch=win64 ?_Tidy@?$_Yarn@D@std@@AEAAXXZ(ptr) _Yarn_char__Tidy
-@ stub -arch=arm ?_Tidy@?$_Yarn@G@std@@AAAXXZ
-@ stub -arch=i386 ?_Tidy@?$_Yarn@G@std@@AAEXXZ
-@ stub -arch=win64 ?_Tidy@?$_Yarn@G@std@@AEAAXXZ
+@ cdecl -arch=arm ?_Tidy@?$_Yarn@G@std@@AAAXXZ(ptr) _Yarn_wchar__Tidy
+@ thiscall -arch=i386 ?_Tidy@?$_Yarn@G@std@@AAEXXZ(ptr) _Yarn_wchar__Tidy
+@ cdecl -arch=win64 ?_Tidy@?$_Yarn@G@std@@AEAAXXZ(ptr) _Yarn_wchar__Tidy
 @ cdecl -arch=arm ?_Tidy@?$_Yarn@_W@std@@AAAXXZ(ptr) _Yarn_wchar__Tidy
 @ thiscall -arch=i386 ?_Tidy@?$_Yarn@_W@std@@AAEXXZ(ptr) _Yarn_wchar__Tidy
 @ cdecl -arch=win64 ?_Tidy@?$_Yarn@_W@std@@AEAAXXZ(ptr) _Yarn_wchar__Tidy
@@ -1719,12 +1719,12 @@
 @ cdecl -arch=arm ?c_str@?$_Yarn@D@std@@QBAPBDXZ(ptr) _Yarn_char_c_str
 @ thiscall -arch=i386 ?c_str@?$_Yarn@D@std@@QBEPBDXZ(ptr) _Yarn_char_c_str
 @ cdecl -arch=win64 ?c_str@?$_Yarn@D@std@@QEBAPEBDXZ(ptr) _Yarn_char_c_str
-@ stub -arch=arm ?c_str@?$_Yarn@G@std@@QBAPBGXZ
-@ stub -arch=i386 ?c_str@?$_Yarn@G@std@@QBEPBGXZ
-@ stub -arch=win64 ?c_str@?$_Yarn@G@std@@QEBAPEBGXZ
-@ stub -arch=arm ?c_str@?$_Yarn@_W@std@@QBAPB_WXZ
-@ stub -arch=i386 ?c_str@?$_Yarn@_W@std@@QBEPB_WXZ
-@ stub -arch=win64 ?c_str@?$_Yarn@_W@std@@QEBAPEB_WXZ
+@ cdecl -arch=arm ?c_str@?$_Yarn@G@std@@QBAPBGXZ(ptr) _Yarn_wchar__C_str
+@ thiscall -arch=i386 ?c_str@?$_Yarn@G@std@@QBEPBGXZ(ptr) _Yarn_wchar__C_str
+@ cdecl -arch=win64 ?c_str@?$_Yarn@G@std@@QEBAPEBGXZ(ptr) _Yarn_wchar__C_str
+@ cdecl -arch=arm ?c_str@?$_Yarn@_W@std@@QBAPB_WXZ(ptr) _Yarn_wchar__C_str
+@ thiscall -arch=i386 ?c_str@?$_Yarn@_W@std@@QBEPB_WXZ(ptr) _Yarn_wchar__C_str
+@ cdecl -arch=win64 ?c_str@?$_Yarn@_W@std@@QEBAPEB_WXZ(ptr) _Yarn_wchar__C_str
 @ extern ?cerr@std@@3V?$basic_ostream@DU?$char_traits@D@std@@@1@A cerr
 @ extern ?cin@std@@3V?$basic_istream@DU?$char_traits@D@std@@@1@A cin
 @ cdecl -arch=win32 ?classic@locale@std@@SAABV12@XZ() locale_classic
@@ -2249,12 +2249,12 @@
 @ cdecl -arch=arm ?empty@?$_Yarn@D@std@@QBA_NXZ(ptr) _Yarn_char_empty
 @ thiscall -arch=i386 ?empty@?$_Yarn@D@std@@QBE_NXZ(ptr) _Yarn_char_empty
 @ cdecl -arch=win64 ?empty@?$_Yarn@D@std@@QEBA_NXZ(ptr) _Yarn_char_empty
-@ stub -arch=arm ?empty@?$_Yarn@G@std@@QBA_NXZ
-@ stub -arch=i386 ?empty@?$_Yarn@G@std@@QBE_NXZ
-@ stub -arch=win64 ?empty@?$_Yarn@G@std@@QEBA_NXZ
-@ stub -arch=arm ?empty@?$_Yarn@_W@std@@QBA_NXZ
-@ stub -arch=i386 ?empty@?$_Yarn@_W@std@@QBE_NXZ
-@ stub -arch=win64 ?empty@?$_Yarn@_W@std@@QEBA_NXZ
+@ cdecl -arch=arm ?empty@?$_Yarn@G@std@@QBA_NXZ(ptr) _Yarn_wchar__Empty
+@ thiscall -arch=i386 ?empty@?$_Yarn@G@std@@QBE_NXZ(ptr) _Yarn_wchar__Empty
+@ cdecl -arch=win64 ?empty@?$_Yarn@G@std@@QEBA_NXZ(ptr) _Yarn_wchar__Empty
+@ cdecl -arch=arm ?empty@?$_Yarn@_W@std@@QBA_NXZ(ptr) _Yarn_wchar__Empty
+@ thiscall -arch=i386 ?empty@?$_Yarn@_W@std@@QBE_NXZ(ptr) _Yarn_wchar__Empty
+@ cdecl -arch=win64 ?empty@?$_Yarn@_W@std@@QEBA_NXZ(ptr) _Yarn_wchar__Empty
 @ cdecl ?empty@locale@std@@SA?AV12@XZ(ptr) locale_empty
 @ cdecl -arch=arm ?encoding@codecvt_base@std@@QBAHXZ(ptr) codecvt_base_encoding
 @ thiscall -arch=i386 ?encoding@codecvt_base@std@@QBEHXZ(ptr) codecvt_base_encoding
diff --git a/dlls/msvcp90/ios.c b/dlls/msvcp90/ios.c
index 2fa7dcb..e83d1b5 100644
--- a/dlls/msvcp90/ios.c
+++ b/dlls/msvcp90/ios.c
@@ -16041,6 +16041,55 @@ WCHAR * __cdecl _Temp_get(WCHAR *dst)
     return dst;
 }
 
+/* ?_Init_cnt@_UShinit@std@@0HA */
+static int _UShinit__Init_cnt = -1;
+
+/* the unsigned short streams' counterpart of _Winit */
+/* ??0_UShinit@std@@QAE@XZ */
+/* ??0_UShinit@std@@QEAA@XZ */
+DEFINE_THISCALL_WRAPPER(_UShinit_ctor, 4)
+void* __thiscall _UShinit_ctor(void *this)
+{
+    TRACE("(%p)\n", this);
+
+    if(_UShinit__Init_cnt < 0)
+        _UShinit__Init_cnt = 1;
+    else
+        _UShinit__Init_cnt++;
+    return this;
+}
+
+/* ??1_UShinit@std@@QAE@XZ */
+/* ??1_UShinit@std@@QEAA@XZ */
+DEFINE_THISCALL_WRAPPER(_UShinit_dtor, 4)
+void __thiscall _UShinit_dtor(void *this)
+{
+    TRACE("(%p)\n", this);
+
+    _UShinit__Init_cnt--;
+    if(!_UShinit__Init_cnt) {
+        basic_ostream_wchar_flush(&ucout.obj);
+        basic_ostream_wchar_flush(&ucerr.obj);
+        basic_ostream_wchar_flush(&uclog.obj);
+    }
+}
+
+/* ??4_UShinit@std@@QAEAAV01@ABV01@@Z */
+/* ??4_UShinit@std@@QEAAAEAV01@AEBV01@@Z */
+/* ??4?$_Iosb@H@std@@QAEAAV01@ABV01@@Z */
+/* ??4?$_Iosb@H@std@@QEAAAEAV01@AEBV01@@Z */
+/* ??4_Init_locks@std@@QAEAAV01@ABV01@@Z */
+/* ??4_Init_locks@std@@QEAAAEAV01@AEBV01@@Z */
+/* ??4_Crt_new_delete@std@@QAEAAU01@ABU01@@Z */
+/* ??4_Crt_new_delete@std@@QEAAAEAU01@AEBU01@@Z */
+/* the assignment of classes without data */
+DEFINE_THISCALL_WRAPPER(empty_class_op_assign, 8)
+void* __thiscall empty_class_op_assign(void *this, const void *rhs)
+{
+    TRACE("(%p %p)\n", this, rhs);
+    return this;
+}
+
 /* ??1_Winit@std@@QAE@XZ */
 /* ??1_Winit@std@@QAE@XZ */
 DEFINE_THISCALL_WRAPPER(_Winit_dtor, 4)
diff --git a/dlls/msvcp90/locale.c b/dlls/msvcp90/locale.c
index 863e55e..18d1802 100644
--- a/dlls/msvcp90/locale.c
+++ b/dlls/msvcp90/locale.c
@@ -405,6 +405,38 @@ void __thiscall locale_facet__Incref(locale_facet *this)
 
 /* ?_Decref@facet@locale@std@@QAEPAV123@XZ */
 /* ?_Decref@facet@locale@std@@QEAAPEAV123@XZ */
+extern const vtable_ptr _Facet_base_vtable;
+
+/* ??0_Facet_base@std@@QAE@XZ */
+/* ??0_Facet_base@std@@QEAA@XZ */
+/* ??0_Facet_base@std@@QAE@ABV01@@Z */
+/* ??0_Facet_base@std@@QEAA@AEBV01@@Z */
+DEFINE_THISCALL_WRAPPER(_Facet_base_ctor, 8)
+void* __thiscall _Facet_base_ctor(void *this, const void *copy)
+{
+    TRACE("(%p %p)\n", this, copy);
+    *(const vtable_ptr **)this = &_Facet_base_vtable;
+    return this;
+}
+
+/* ??1_Facet_base@std@@UAE@XZ */
+/* ??1_Facet_base@std@@UEAA@XZ */
+DEFINE_THISCALL_WRAPPER(_Facet_base_dtor, 4)
+void __thiscall _Facet_base_dtor(void *this)
+{
+    TRACE("(%p)\n", this);
+}
+
+/* ??4_Facet_base@std@@QAEAAV01@ABV01@@Z */
+/* ??4_Facet_base@std@@QEAAAEAV01@AEBV01@@Z */
+/* the vtable stays the object's own */
+DEFINE_THISCALL_WRAPPER(_Facet_base_op_assign, 8)
+void* __thiscall _Facet_base_op_assign(void *this, const void *rhs)
+{
+    TRACE("(%p %p)\n", this, rhs);
+    return this;
+}
+
 /* ?_Decref@facet@locale@std@@UAEPAV_Facet_base@3@XZ */
 /* ?_Decref@facet@locale@std@@UEAAPEAV_Facet_base@3@XZ */
 #if _MSVCP_VER >= 110
diff --git a/dlls/msvcp90/string.c b/dlls/msvcp90/string.c
index 10fae25..b6e70f8 100644
--- a/dlls/msvcp90/string.c
+++ b/dlls/msvcp90/string.c
@@ -1763,6 +1763,42 @@ _Yarn_wchar* __thiscall _Yarn_wchar_op_assign_cstr(_Yarn_wchar *this, const wcha
     return this;
 }
 
+/* ??0?$_Yarn@_W@std@@QAE@PB_W@Z */
+/* ??0?$_Yarn@_W@std@@QEAA@PEB_W@Z */
+DEFINE_THISCALL_WRAPPER(_Yarn_wchar_ctor_cstr, 8)
+_Yarn_wchar* __thiscall _Yarn_wchar_ctor_cstr(_Yarn_wchar *this, const wchar_t *str)
+{
+    TRACE("(%p %p)\n", this, str);
+
+    _Yarn_wchar_ctor(this);
+    return _Yarn_wchar_op_assign_cstr(this, str);
+}
+
+/* ??4?$_Yarn@_W@std@@QAEAAV01@ABV01@@Z */
+/* ??4?$_Yarn@_W@std@@QEAAAEAV01@AEBV01@@Z */
+DEFINE_THISCALL_WRAPPER(_Yarn_wchar_op_assign, 8)
+_Yarn_wchar* __thiscall _Yarn_wchar_op_assign(_Yarn_wchar *this, const _Yarn_wchar *rhs)
+{
+    TRACE("(%p %p)\n", this, rhs);
+
+#ifdef SG_MUTANT_YARN_ASSIGN_SHARES
+    this->str = rhs->str;
+    return this;
+#endif
+    return _Yarn_wchar_op_assign_cstr(this, rhs->str);
+}
+
+/* ??0?$_Yarn@_W@std@@QAE@ABV01@@Z */
+/* ??0?$_Yarn@_W@std@@QEAA@AEBV01@@Z */
+DEFINE_THISCALL_WRAPPER(_Yarn_wchar_copy_ctor, 8)
+_Yarn_wchar* __thiscall _Yarn_wchar_copy_ctor(_Yarn_wchar *this, const _Yarn_wchar *copy)
+{
+    TRACE("(%p %p)\n", this, copy);
+
+    _Yarn_wchar_ctor(this);
+    return _Yarn_wchar_op_assign(this, copy);
+}
+
 /* ?_C_str@?$_Yarn@_W@std@@QBEPB_WXZ */
 /* ?_C_str@?$_Yarn@_W@std@@QEBAPEB_WXZ */
 DEFINE_THISCALL_WRAPPER(_Yarn_wchar__C_str, 4)
diff --git a/dlls/msvcp_win/msvcp_win.spec b/dlls/msvcp_win/msvcp_win.spec
index f204c3c..87e4004 100644
--- a/dlls/msvcp_win/msvcp_win.spec
+++ b/dlls/msvcp_win/msvcp_win.spec
@@ -16,21 +16,21 @@
 @ cdecl -arch=arm ??0?$_Yarn@D@std@@QAA@XZ(ptr) msvcp140.??0?$_Yarn@D@std@@QAA@XZ
 @ thiscall -arch=i386 ??0?$_Yarn@D@std@@QAE@XZ(ptr) msvcp140.??0?$_Yarn@D@std@@QAE@XZ
 @ cdecl -arch=win64 ??0?$_Yarn@D@std@@QEAA@XZ(ptr) msvcp140.??0?$_Yarn@D@std@@QEAA@XZ
-@ stub -arch=arm ??0?$_Yarn@G@std@@QAA@ABV01@@Z
-@ stub -arch=i386 ??0?$_Yarn@G@std@@QAE@ABV01@@Z
-@ stub -arch=win64 ??0?$_Yarn@G@std@@QEAA@AEBV01@@Z
-@ stub -arch=arm ??0?$_Yarn@G@std@@QAA@PBG@Z
-@ stub -arch=i386 ??0?$_Yarn@G@std@@QAE@PBG@Z
-@ stub -arch=win64 ??0?$_Yarn@G@std@@QEAA@PEBG@Z
-@ stub -arch=arm ??0?$_Yarn@G@std@@QAA@XZ
-@ stub -arch=i386 ??0?$_Yarn@G@std@@QAE@XZ
-@ stub -arch=win64 ??0?$_Yarn@G@std@@QEAA@XZ
-@ stub -arch=arm ??0?$_Yarn@_W@std@@QAA@ABV01@@Z
-@ stub -arch=i386 ??0?$_Yarn@_W@std@@QAE@ABV01@@Z
-@ stub -arch=win64 ??0?$_Yarn@_W@std@@QEAA@AEBV01@@Z
-@ stub -arch=arm ??0?$_Yarn@_W@std@@QAA@PB_W@Z
-@ stub -arch=i386 ??0?$_Yarn@_W@std@@QAE@PB_W@Z
-@ stub -arch=win64 ??0?$_Yarn@_W@std@@QEAA@PEB_W@Z
+@ cdecl -arch=arm ??0?$_Yarn@G@std@@QAA@ABV01@@Z(ptr ptr) msvcp140.??0?$_Yarn@G@std@@QAA@ABV01@@Z
+@ thiscall -arch=i386 ??0?$_Yarn@G@std@@QAE@ABV01@@Z(ptr ptr) msvcp140.??0?$_Yarn@G@std@@QAE@ABV01@@Z
+@ cdecl -arch=win64 ??0?$_Yarn@G@std@@QEAA@AEBV01@@Z(ptr ptr) msvcp140.??0?$_Yarn@G@std@@QEAA@AEBV01@@Z
+@ cdecl -arch=arm ??0?$_Yarn@G@std@@QAA@PBG@Z(ptr wstr) msvcp140.??0?$_Yarn@G@std@@QAA@PBG@Z
+@ thiscall -arch=i386 ??0?$_Yarn@G@std@@QAE@PBG@Z(ptr wstr) msvcp140.??0?$_Yarn@G@std@@QAE@PBG@Z
+@ cdecl -arch=win64 ??0?$_Yarn@G@std@@QEAA@PEBG@Z(ptr wstr) msvcp140.??0?$_Yarn@G@std@@QEAA@PEBG@Z
+@ cdecl -arch=arm ??0?$_Yarn@G@std@@QAA@XZ(ptr) msvcp140.??0?$_Yarn@G@std@@QAA@XZ
+@ thiscall -arch=i386 ??0?$_Yarn@G@std@@QAE@XZ(ptr) msvcp140.??0?$_Yarn@G@std@@QAE@XZ
+@ cdecl -arch=win64 ??0?$_Yarn@G@std@@QEAA@XZ(ptr) msvcp140.??0?$_Yarn@G@std@@QEAA@XZ
+@ cdecl -arch=arm ??0?$_Yarn@_W@std@@QAA@ABV01@@Z(ptr ptr) msvcp140.??0?$_Yarn@_W@std@@QAA@ABV01@@Z
+@ thiscall -arch=i386 ??0?$_Yarn@_W@std@@QAE@ABV01@@Z(ptr ptr) msvcp140.??0?$_Yarn@_W@std@@QAE@ABV01@@Z
+@ cdecl -arch=win64 ??0?$_Yarn@_W@std@@QEAA@AEBV01@@Z(ptr ptr) msvcp140.??0?$_Yarn@_W@std@@QEAA@AEBV01@@Z
+@ cdecl -arch=arm ??0?$_Yarn@_W@std@@QAA@PB_W@Z(ptr wstr) msvcp140.??0?$_Yarn@_W@std@@QAA@PB_W@Z
+@ thiscall -arch=i386 ??0?$_Yarn@_W@std@@QAE@PB_W@Z(ptr wstr) msvcp140.??0?$_Yarn@_W@std@@QAE@PB_W@Z
+@ cdecl -arch=win64 ??0?$_Yarn@_W@std@@QEAA@PEB_W@Z(ptr wstr) msvcp140.??0?$_Yarn@_W@std@@QEAA@PEB_W@Z
 @ cdecl -arch=arm ??0?$_Yarn@_W@std@@QAA@XZ(ptr) msvcp140.??0?$_Yarn@_W@std@@QAA@XZ
 @ thiscall -arch=i386 ??0?$_Yarn@_W@std@@QAE@XZ(ptr) msvcp140.??0?$_Yarn@_W@std@@QAE@XZ
 @ cdecl -arch=win64 ??0?$_Yarn@_W@std@@QEAA@XZ(ptr) msvcp140.??0?$_Yarn@_W@std@@QEAA@XZ
@@ -304,12 +304,12 @@
 @ cdecl -arch=arm ??0Init@ios_base@std@@QAA@XZ(ptr) msvcp140.??0Init@ios_base@std@@QAA@XZ
 @ thiscall -arch=i386 ??0Init@ios_base@std@@QAE@XZ(ptr) msvcp140.??0Init@ios_base@std@@QAE@XZ
 @ cdecl -arch=win64 ??0Init@ios_base@std@@QEAA@XZ(ptr) msvcp140.??0Init@ios_base@std@@QEAA@XZ
-@ stub -arch=arm ??0_Facet_base@std@@QAA@ABV01@@Z
-@ stub -arch=i386 ??0_Facet_base@std@@QAE@ABV01@@Z
-@ stub -arch=win64 ??0_Facet_base@std@@QEAA@AEBV01@@Z
-@ stub -arch=arm ??0_Facet_base@std@@QAA@XZ
-@ stub -arch=i386 ??0_Facet_base@std@@QAE@XZ
-@ stub -arch=win64 ??0_Facet_base@std@@QEAA@XZ
+@ cdecl -arch=arm ??0_Facet_base@std@@QAA@ABV01@@Z(ptr ptr) msvcp140.??0_Facet_base@std@@QAA@ABV01@@Z
+@ thiscall -arch=i386 ??0_Facet_base@std@@QAE@ABV01@@Z(ptr ptr) msvcp140.??0_Facet_base@std@@QAE@ABV01@@Z
+@ cdecl -arch=win64 ??0_Facet_base@std@@QEAA@AEBV01@@Z(ptr ptr) msvcp140.??0_Facet_base@std@@QEAA@AEBV01@@Z
+@ cdecl -arch=arm ??0_Facet_base@std@@QAA@XZ(ptr ptr) msvcp140.??0_Facet_base@std@@QAA@XZ
+@ thiscall -arch=i386 ??0_Facet_base@std@@QAE@XZ(ptr ptr) msvcp140.??0_Facet_base@std@@QAE@XZ
+@ cdecl -arch=win64 ??0_Facet_base@std@@QEAA@XZ(ptr ptr) msvcp140.??0_Facet_base@std@@QEAA@XZ
 @ cdecl -arch=arm ??0_Init_locks@std@@QAA@XZ(ptr) msvcp140.??0_Init_locks@std@@QAA@XZ
 @ thiscall -arch=i386 ??0_Init_locks@std@@QAE@XZ(ptr) msvcp140.??0_Init_locks@std@@QAE@XZ
 @ cdecl -arch=win64 ??0_Init_locks@std@@QEAA@XZ(ptr) msvcp140.??0_Init_locks@std@@QEAA@XZ
@@ -337,9 +337,9 @@
 @ cdecl -arch=arm ??0_Timevec@std@@QAA@PAX@Z(ptr ptr) msvcp140.??0_Timevec@std@@QAA@PAX@Z
 @ thiscall -arch=i386 ??0_Timevec@std@@QAE@PAX@Z(ptr ptr) msvcp140.??0_Timevec@std@@QAE@PAX@Z
 @ cdecl -arch=win64 ??0_Timevec@std@@QEAA@PEAX@Z(ptr ptr) msvcp140.??0_Timevec@std@@QEAA@PEAX@Z
-@ stub -arch=arm ??0_UShinit@std@@QAA@XZ
-@ stub -arch=i386 ??0_UShinit@std@@QAE@XZ
-@ stub -arch=win64 ??0_UShinit@std@@QEAA@XZ
+@ cdecl -arch=arm ??0_UShinit@std@@QAA@XZ(ptr) msvcp140.??0_UShinit@std@@QAA@XZ
+@ thiscall -arch=i386 ??0_UShinit@std@@QAE@XZ(ptr) msvcp140.??0_UShinit@std@@QAE@XZ
+@ cdecl -arch=win64 ??0_UShinit@std@@QEAA@XZ(ptr) msvcp140.??0_UShinit@std@@QEAA@XZ
 @ cdecl -arch=arm ??0_Winit@std@@QAA@XZ(ptr) msvcp140.??0_Winit@std@@QAA@XZ
 @ thiscall -arch=i386 ??0_Winit@std@@QAE@XZ(ptr) msvcp140.??0_Winit@std@@QAE@XZ
 @ cdecl -arch=win64 ??0_Winit@std@@QEAA@XZ(ptr) msvcp140.??0_Winit@std@@QEAA@XZ
@@ -367,9 +367,9 @@
 @ cdecl -arch=arm ??1?$_Yarn@D@std@@QAA@XZ(ptr) msvcp140.??1?$_Yarn@D@std@@QAA@XZ
 @ thiscall -arch=i386 ??1?$_Yarn@D@std@@QAE@XZ(ptr) msvcp140.??1?$_Yarn@D@std@@QAE@XZ
 @ cdecl -arch=win64 ??1?$_Yarn@D@std@@QEAA@XZ(ptr) msvcp140.??1?$_Yarn@D@std@@QEAA@XZ
-@ stub -arch=arm ??1?$_Yarn@G@std@@QAA@XZ
-@ stub -arch=i386 ??1?$_Yarn@G@std@@QAE@XZ
-@ stub -arch=win64 ??1?$_Yarn@G@std@@QEAA@XZ
+@ cdecl -arch=arm ??1?$_Yarn@G@std@@QAA@XZ(ptr) msvcp140.??1?$_Yarn@G@std@@QAA@XZ
+@ thiscall -arch=i386 ??1?$_Yarn@G@std@@QAE@XZ(ptr) msvcp140.??1?$_Yarn@G@std@@QAE@XZ
+@ cdecl -arch=win64 ??1?$_Yarn@G@std@@QEAA@XZ(ptr) msvcp140.??1?$_Yarn@G@std@@QEAA@XZ
 @ cdecl -arch=arm ??1?$_Yarn@_W@std@@QAA@XZ(ptr) msvcp140.??1?$_Yarn@_W@std@@QAA@XZ
 @ thiscall -arch=i386 ??1?$_Yarn@_W@std@@QAE@XZ(ptr) msvcp140.??1?$_Yarn@_W@std@@QAE@XZ
 @ cdecl -arch=win64 ??1?$_Yarn@_W@std@@QEAA@XZ(ptr) msvcp140.??1?$_Yarn@_W@std@@QEAA@XZ
@@ -481,9 +481,9 @@
 @ cdecl -arch=arm ??1Init@ios_base@std@@QAA@XZ(ptr) msvcp140.??1Init@ios_base@std@@QAA@XZ
 @ thiscall -arch=i386 ??1Init@ios_base@std@@QAE@XZ(ptr) msvcp140.??1Init@ios_base@std@@QAE@XZ
 @ cdecl -arch=win64 ??1Init@ios_base@std@@QEAA@XZ(ptr) msvcp140.??1Init@ios_base@std@@QEAA@XZ
-@ stub -arch=arm ??1_Facet_base@std@@UAA@XZ
-@ stub -arch=i386 ??1_Facet_base@std@@UAE@XZ
-@ stub -arch=win64 ??1_Facet_base@std@@UEAA@XZ
+@ cdecl -arch=arm ??1_Facet_base@std@@UAA@XZ(ptr) msvcp140.??1_Facet_base@std@@UAA@XZ
+@ thiscall -arch=i386 ??1_Facet_base@std@@UAE@XZ(ptr) msvcp140.??1_Facet_base@std@@UAE@XZ
+@ cdecl -arch=win64 ??1_Facet_base@std@@UEAA@XZ(ptr) msvcp140.??1_Facet_base@std@@UEAA@XZ
 @ cdecl -arch=arm ??1_Init_locks@std@@QAA@XZ(ptr) msvcp140.??1_Init_locks@std@@QAA@XZ
 @ thiscall -arch=i386 ??1_Init_locks@std@@QAE@XZ(ptr) msvcp140.??1_Init_locks@std@@QAE@XZ
 @ cdecl -arch=win64 ??1_Init_locks@std@@QEAA@XZ(ptr) msvcp140.??1_Init_locks@std@@QEAA@XZ
@@ -499,9 +499,9 @@
 @ cdecl -arch=arm ??1_Timevec@std@@QAA@XZ(ptr) msvcp140.??1_Timevec@std@@QAA@XZ
 @ thiscall -arch=i386 ??1_Timevec@std@@QAE@XZ(ptr) msvcp140.??1_Timevec@std@@QAE@XZ
 @ cdecl -arch=win64 ??1_Timevec@std@@QEAA@XZ(ptr) msvcp140.??1_Timevec@std@@QEAA@XZ
-@ stub -arch=arm ??1_UShinit@std@@QAA@XZ
-@ stub -arch=i386 ??1_UShinit@std@@QAE@XZ
-@ stub -arch=win64 ??1_UShinit@std@@QEAA@XZ
+@ cdecl -arch=arm ??1_UShinit@std@@QAA@XZ(ptr) msvcp140.??1_UShinit@std@@QAA@XZ
+@ thiscall -arch=i386 ??1_UShinit@std@@QAE@XZ(ptr) msvcp140.??1_UShinit@std@@QAE@XZ
+@ cdecl -arch=win64 ??1_UShinit@std@@QEAA@XZ(ptr) msvcp140.??1_UShinit@std@@QEAA@XZ
 @ cdecl -arch=arm ??1_Winit@std@@QAA@XZ(ptr) msvcp140.??1_Winit@std@@QAA@XZ
 @ thiscall -arch=i386 ??1_Winit@std@@QAE@XZ(ptr) msvcp140.??1_Winit@std@@QAE@XZ
 @ cdecl -arch=win64 ??1_Winit@std@@QEAA@XZ(ptr) msvcp140.??1_Winit@std@@QEAA@XZ
@@ -520,27 +520,27 @@
 @ cdecl -arch=arm ??1time_base@std@@UAA@XZ(ptr) msvcp140.??1time_base@std@@UAA@XZ
 @ thiscall -arch=i386 ??1time_base@std@@UAE@XZ(ptr) msvcp140.??1time_base@std@@UAE@XZ
 @ cdecl -arch=win64 ??1time_base@std@@UEAA@XZ(ptr) msvcp140.??1time_base@std@@UEAA@XZ
-@ stub -arch=arm ??4?$_Iosb@H@std@@QAAAAV01@$$QAV01@@Z
-@ stub -arch=i386 ??4?$_Iosb@H@std@@QAEAAV01@$$QAV01@@Z
+@ cdecl -arch=arm ??4?$_Iosb@H@std@@QAAAAV01@$$QAV01@@Z(ptr ptr) msvcp140.??4?$_Iosb@H@std@@QAAAAV01@$$QAV01@@Z
+@ thiscall -arch=i386 ??4?$_Iosb@H@std@@QAEAAV01@$$QAV01@@Z(ptr ptr) msvcp140.??4?$_Iosb@H@std@@QAEAAV01@$$QAV01@@Z
 @ stub -arch=win64 ??4?$_Iosb@H@std@@QEAAAEAV01@$$QEAV01@@Z
-@ stub -arch=arm ??4?$_Iosb@H@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Iosb@H@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Iosb@H@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4?$_Iosb@H@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp140.??4?$_Iosb@H@std@@QAAAAV01@ABV01@@Z
+@ thiscall -arch=i386 ??4?$_Iosb@H@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp140.??4?$_Iosb@H@std@@QAEAAV01@ABV01@@Z
+@ cdecl -arch=win64 ??4?$_Iosb@H@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp140.??4?$_Iosb@H@std@@QEAAAEAV01@AEBV01@@Z
 @ cdecl -arch=arm ??4?$_Yarn@D@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp140.??4?$_Yarn@D@std@@QAAAAV01@ABV01@@Z
 @ thiscall -arch=i386 ??4?$_Yarn@D@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp140.??4?$_Yarn@D@std@@QAEAAV01@ABV01@@Z
 @ cdecl -arch=win64 ??4?$_Yarn@D@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp140.??4?$_Yarn@D@std@@QEAAAEAV01@AEBV01@@Z
 @ cdecl -arch=arm ??4?$_Yarn@D@std@@QAAAAV01@PBD@Z(ptr str) msvcp140.??4?$_Yarn@D@std@@QAAAAV01@PBD@Z
 @ thiscall -arch=i386 ??4?$_Yarn@D@std@@QAEAAV01@PBD@Z(ptr str) msvcp140.??4?$_Yarn@D@std@@QAEAAV01@PBD@Z
 @ cdecl -arch=win64 ??4?$_Yarn@D@std@@QEAAAEAV01@PEBD@Z(ptr str) msvcp140.??4?$_Yarn@D@std@@QEAAAEAV01@PEBD@Z
-@ stub -arch=arm ??4?$_Yarn@G@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Yarn@G@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Yarn@G@std@@QEAAAEAV01@AEBV01@@Z
-@ stub -arch=arm ??4?$_Yarn@G@std@@QAAAAV01@PBG@Z
-@ stub -arch=i386 ??4?$_Yarn@G@std@@QAEAAV01@PBG@Z
-@ stub -arch=win64 ??4?$_Yarn@G@std@@QEAAAEAV01@PEBG@Z
-@ stub -arch=arm ??4?$_Yarn@_W@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Yarn@_W@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Yarn@_W@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4?$_Yarn@G@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp140.??4?$_Yarn@G@std@@QAAAAV01@ABV01@@Z
+@ thiscall -arch=i386 ??4?$_Yarn@G@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp140.??4?$_Yarn@G@std@@QAEAAV01@ABV01@@Z
+@ cdecl -arch=win64 ??4?$_Yarn@G@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp140.??4?$_Yarn@G@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4?$_Yarn@G@std@@QAAAAV01@PBG@Z(ptr wstr) msvcp140.??4?$_Yarn@G@std@@QAAAAV01@PBG@Z
+@ thiscall -arch=i386 ??4?$_Yarn@G@std@@QAEAAV01@PBG@Z(ptr wstr) msvcp140.??4?$_Yarn@G@std@@QAEAAV01@PBG@Z
+@ cdecl -arch=win64 ??4?$_Yarn@G@std@@QEAAAEAV01@PEBG@Z(ptr wstr) msvcp140.??4?$_Yarn@G@std@@QEAAAEAV01@PEBG@Z
+@ cdecl -arch=arm ??4?$_Yarn@_W@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp140.??4?$_Yarn@_W@std@@QAAAAV01@ABV01@@Z
+@ thiscall -arch=i386 ??4?$_Yarn@_W@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp140.??4?$_Yarn@_W@std@@QAEAAV01@ABV01@@Z
+@ cdecl -arch=win64 ??4?$_Yarn@_W@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp140.??4?$_Yarn@_W@std@@QEAAAEAV01@AEBV01@@Z
 @ cdecl -arch=arm ??4?$_Yarn@_W@std@@QAAAAV01@PB_W@Z(ptr wstr) msvcp140.??4?$_Yarn@_W@std@@QAAAAV01@PB_W@Z
 @ thiscall -arch=i386 ??4?$_Yarn@_W@std@@QAEAAV01@PB_W@Z(ptr wstr) msvcp140.??4?$_Yarn@_W@std@@QAEAAV01@PB_W@Z
 @ cdecl -arch=win64 ??4?$_Yarn@_W@std@@QEAAAEAV01@PEB_W@Z(ptr wstr) msvcp140.??4?$_Yarn@_W@std@@QEAAAEAV01@PEB_W@Z
@@ -583,24 +583,24 @@
 @ cdecl -arch=arm ??4Init@ios_base@std@@QAAAAV012@ABV012@@Z(ptr ptr) msvcp140.??4Init@ios_base@std@@QAAAAV012@ABV012@@Z
 @ thiscall -arch=i386 ??4Init@ios_base@std@@QAEAAV012@ABV012@@Z(ptr ptr) msvcp140.??4Init@ios_base@std@@QAEAAV012@ABV012@@Z
 @ cdecl -arch=win64 ??4Init@ios_base@std@@QEAAAEAV012@AEBV012@@Z(ptr ptr) msvcp140.??4Init@ios_base@std@@QEAAAEAV012@AEBV012@@Z
-@ stub -arch=arm ??4_Crt_new_delete@std@@QAAAAU01@$$QAU01@@Z
-@ stub -arch=i386 ??4_Crt_new_delete@std@@QAEAAU01@$$QAU01@@Z
-@ stub -arch=win64 ??4_Crt_new_delete@std@@QEAAAEAU01@$$QEAU01@@Z
-@ stub -arch=arm ??4_Crt_new_delete@std@@QAAAAU01@ABU01@@Z
-@ stub -arch=i386 ??4_Crt_new_delete@std@@QAEAAU01@ABU01@@Z
-@ stub -arch=win64 ??4_Crt_new_delete@std@@QEAAAEAU01@AEBU01@@Z
-@ stub -arch=arm ??4_Facet_base@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_Facet_base@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_Facet_base@std@@QEAAAEAV01@AEBV01@@Z
-@ stub -arch=arm ??4_Init_locks@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_Init_locks@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_Init_locks@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4_Crt_new_delete@std@@QAAAAU01@$$QAU01@@Z(ptr ptr) msvcp140.??4_Crt_new_delete@std@@QAAAAU01@$$QAU01@@Z
+@ thiscall -arch=i386 ??4_Crt_new_delete@std@@QAEAAU01@$$QAU01@@Z(ptr ptr) msvcp140.??4_Crt_new_delete@std@@QAEAAU01@$$QAU01@@Z
+@ cdecl -arch=win64 ??4_Crt_new_delete@std@@QEAAAEAU01@$$QEAU01@@Z(ptr ptr) msvcp140.??4_Crt_new_delete@std@@QEAAAEAU01@$$QEAU01@@Z
+@ cdecl -arch=arm ??4_Crt_new_delete@std@@QAAAAU01@ABU01@@Z(ptr ptr) msvcp140.??4_Crt_new_delete@std@@QAAAAU01@ABU01@@Z
+@ thiscall -arch=i386 ??4_Crt_new_delete@std@@QAEAAU01@ABU01@@Z(ptr ptr) msvcp140.??4_Crt_new_delete@std@@QAEAAU01@ABU01@@Z
+@ cdecl -arch=win64 ??4_Crt_new_delete@std@@QEAAAEAU01@AEBU01@@Z(ptr ptr) msvcp140.??4_Crt_new_delete@std@@QEAAAEAU01@AEBU01@@Z
+@ cdecl -arch=arm ??4_Facet_base@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp140.??4_Facet_base@std@@QAAAAV01@ABV01@@Z
+@ thiscall -arch=i386 ??4_Facet_base@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp140.??4_Facet_base@std@@QAEAAV01@ABV01@@Z
+@ cdecl -arch=win64 ??4_Facet_base@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp140.??4_Facet_base@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4_Init_locks@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp140.??4_Init_locks@std@@QAAAAV01@ABV01@@Z
+@ thiscall -arch=i386 ??4_Init_locks@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp140.??4_Init_locks@std@@QAEAAV01@ABV01@@Z
+@ cdecl -arch=win64 ??4_Init_locks@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp140.??4_Init_locks@std@@QEAAAEAV01@AEBV01@@Z
 @ cdecl -arch=arm ??4_Timevec@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp140.??4_Timevec@std@@QAAAAV01@ABV01@@Z
 @ thiscall -arch=i386 ??4_Timevec@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp140.??4_Timevec@std@@QAEAAV01@ABV01@@Z
 @ cdecl -arch=win64 ??4_Timevec@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp140.??4_Timevec@std@@QEAAAEAV01@AEBV01@@Z
-@ stub -arch=arm ??4_UShinit@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4_UShinit@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4_UShinit@std@@QEAAAEAV01@AEBV01@@Z
+@ cdecl -arch=arm ??4_UShinit@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp140.??4_UShinit@std@@QAAAAV01@ABV01@@Z
+@ thiscall -arch=i386 ??4_UShinit@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp140.??4_UShinit@std@@QAEAAV01@ABV01@@Z
+@ cdecl -arch=win64 ??4_UShinit@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp140.??4_UShinit@std@@QEAAAEAV01@AEBV01@@Z
 @ cdecl -arch=arm ??4_Winit@std@@QAAAAV01@ABV01@@Z(ptr ptr) msvcp140.??4_Winit@std@@QAAAAV01@ABV01@@Z
 @ thiscall -arch=i386 ??4_Winit@std@@QAEAAV01@ABV01@@Z(ptr ptr) msvcp140.??4_Winit@std@@QAEAAV01@ABV01@@Z
 @ cdecl -arch=win64 ??4_Winit@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) msvcp140.??4_Winit@std@@QEAAAEAV01@AEBV01@@Z
@@ -1104,9 +1104,9 @@
 @ cdecl -arch=arm ?_C_str@?$_Yarn@D@std@@QBAPBDXZ(ptr) msvcp140.?_C_str@?$_Yarn@D@std@@QBAPBDXZ
 @ thiscall -arch=i386 ?_C_str@?$_Yarn@D@std@@QBEPBDXZ(ptr) msvcp140.?_C_str@?$_Yarn@D@std@@QBEPBDXZ
 @ cdecl -arch=win64 ?_C_str@?$_Yarn@D@std@@QEBAPEBDXZ(ptr) msvcp140.?_C_str@?$_Yarn@D@std@@QEBAPEBDXZ
-@ stub -arch=arm ?_C_str@?$_Yarn@G@std@@QBAPBGXZ
-@ stub -arch=i386 ?_C_str@?$_Yarn@G@std@@QBEPBGXZ
-@ stub -arch=win64 ?_C_str@?$_Yarn@G@std@@QEBAPEBGXZ
+@ cdecl -arch=arm ?_C_str@?$_Yarn@G@std@@QBAPBGXZ(ptr) msvcp140.?_C_str@?$_Yarn@G@std@@QBAPBGXZ
+@ thiscall -arch=i386 ?_C_str@?$_Yarn@G@std@@QBEPBGXZ(ptr) msvcp140.?_C_str@?$_Yarn@G@std@@QBEPBGXZ
+@ cdecl -arch=win64 ?_C_str@?$_Yarn@G@std@@QEBAPEBGXZ(ptr) msvcp140.?_C_str@?$_Yarn@G@std@@QEBAPEBGXZ
 @ cdecl -arch=arm ?_C_str@?$_Yarn@_W@std@@QBAPB_WXZ(ptr) msvcp140.?_C_str@?$_Yarn@_W@std@@QBAPB_WXZ
 @ thiscall -arch=i386 ?_C_str@?$_Yarn@_W@std@@QBEPB_WXZ(ptr) msvcp140.?_C_str@?$_Yarn@_W@std@@QBEPB_WXZ
 @ cdecl -arch=win64 ?_C_str@?$_Yarn@_W@std@@QEBAPEB_WXZ(ptr) msvcp140.?_C_str@?$_Yarn@_W@std@@QEBAPEB_WXZ
@@ -1139,9 +1139,9 @@
 @ cdecl -arch=arm ?_Empty@?$_Yarn@D@std@@QBA_NXZ(ptr) msvcp140.?_Empty@?$_Yarn@D@std@@QBA_NXZ
 @ thiscall -arch=i386 ?_Empty@?$_Yarn@D@std@@QBE_NXZ(ptr) msvcp140.?_Empty@?$_Yarn@D@std@@QBE_NXZ
 @ cdecl -arch=win64 ?_Empty@?$_Yarn@D@std@@QEBA_NXZ(ptr) msvcp140.?_Empty@?$_Yarn@D@std@@QEBA_NXZ
-@ stub -arch=arm ?_Empty@?$_Yarn@G@std@@QBA_NXZ
-@ stub -arch=i386 ?_Empty@?$_Yarn@G@std@@QBE_NXZ
-@ stub -arch=win64 ?_Empty@?$_Yarn@G@std@@QEBA_NXZ
+@ cdecl -arch=arm ?_Empty@?$_Yarn@G@std@@QBA_NXZ(ptr) msvcp140.?_Empty@?$_Yarn@G@std@@QBA_NXZ
+@ thiscall -arch=i386 ?_Empty@?$_Yarn@G@std@@QBE_NXZ(ptr) msvcp140.?_Empty@?$_Yarn@G@std@@QBE_NXZ
+@ cdecl -arch=win64 ?_Empty@?$_Yarn@G@std@@QEBA_NXZ(ptr) msvcp140.?_Empty@?$_Yarn@G@std@@QEBA_NXZ
 @ cdecl -arch=arm ?_Empty@?$_Yarn@_W@std@@QBA_NXZ(ptr) msvcp140.?_Empty@?$_Yarn@_W@std@@QBA_NXZ
 @ thiscall -arch=i386 ?_Empty@?$_Yarn@_W@std@@QBE_NXZ(ptr) msvcp140.?_Empty@?$_Yarn@_W@std@@QBE_NXZ
 @ cdecl -arch=win64 ?_Empty@?$_Yarn@_W@std@@QEBA_NXZ(ptr) msvcp140.?_Empty@?$_Yarn@_W@std@@QEBA_NXZ
@@ -1630,9 +1630,9 @@
 @ cdecl -arch=arm ?_Tidy@?$_Yarn@D@std@@AAAXXZ(ptr) msvcp140.?_Tidy@?$_Yarn@D@std@@AAAXXZ
 @ thiscall -arch=i386 ?_Tidy@?$_Yarn@D@std@@AAEXXZ(ptr) msvcp140.?_Tidy@?$_Yarn@D@std@@AAEXXZ
 @ cdecl -arch=win64 ?_Tidy@?$_Yarn@D@std@@AEAAXXZ(ptr) msvcp140.?_Tidy@?$_Yarn@D@std@@AEAAXXZ
-@ stub -arch=arm ?_Tidy@?$_Yarn@G@std@@AAAXXZ
-@ stub -arch=i386 ?_Tidy@?$_Yarn@G@std@@AAEXXZ
-@ stub -arch=win64 ?_Tidy@?$_Yarn@G@std@@AEAAXXZ
+@ cdecl -arch=arm ?_Tidy@?$_Yarn@G@std@@AAAXXZ(ptr) msvcp140.?_Tidy@?$_Yarn@G@std@@AAAXXZ
+@ thiscall -arch=i386 ?_Tidy@?$_Yarn@G@std@@AAEXXZ(ptr) msvcp140.?_Tidy@?$_Yarn@G@std@@AAEXXZ
+@ cdecl -arch=win64 ?_Tidy@?$_Yarn@G@std@@AEAAXXZ(ptr) msvcp140.?_Tidy@?$_Yarn@G@std@@AEAAXXZ
 @ cdecl -arch=arm ?_Tidy@?$_Yarn@_W@std@@AAAXXZ(ptr) msvcp140.?_Tidy@?$_Yarn@_W@std@@AAAXXZ
 @ thiscall -arch=i386 ?_Tidy@?$_Yarn@_W@std@@AAEXXZ(ptr) msvcp140.?_Tidy@?$_Yarn@_W@std@@AAEXXZ
 @ cdecl -arch=win64 ?_Tidy@?$_Yarn@_W@std@@AEAAXXZ(ptr) msvcp140.?_Tidy@?$_Yarn@_W@std@@AEAAXXZ
@@ -1718,12 +1718,12 @@
 @ cdecl -arch=arm ?c_str@?$_Yarn@D@std@@QBAPBDXZ(ptr) msvcp140.?c_str@?$_Yarn@D@std@@QBAPBDXZ
 @ thiscall -arch=i386 ?c_str@?$_Yarn@D@std@@QBEPBDXZ(ptr) msvcp140.?c_str@?$_Yarn@D@std@@QBEPBDXZ
 @ cdecl -arch=win64 ?c_str@?$_Yarn@D@std@@QEBAPEBDXZ(ptr) msvcp140.?c_str@?$_Yarn@D@std@@QEBAPEBDXZ
-@ stub -arch=arm ?c_str@?$_Yarn@G@std@@QBAPBGXZ
-@ stub -arch=i386 ?c_str@?$_Yarn@G@std@@QBEPBGXZ
-@ stub -arch=win64 ?c_str@?$_Yarn@G@std@@QEBAPEBGXZ
-@ stub -arch=arm ?c_str@?$_Yarn@_W@std@@QBAPB_WXZ
-@ stub -arch=i386 ?c_str@?$_Yarn@_W@std@@QBEPB_WXZ
-@ stub -arch=win64 ?c_str@?$_Yarn@_W@std@@QEBAPEB_WXZ
+@ cdecl -arch=arm ?c_str@?$_Yarn@G@std@@QBAPBGXZ(ptr) msvcp140.?c_str@?$_Yarn@G@std@@QBAPBGXZ
+@ thiscall -arch=i386 ?c_str@?$_Yarn@G@std@@QBEPBGXZ(ptr) msvcp140.?c_str@?$_Yarn@G@std@@QBEPBGXZ
+@ cdecl -arch=win64 ?c_str@?$_Yarn@G@std@@QEBAPEBGXZ(ptr) msvcp140.?c_str@?$_Yarn@G@std@@QEBAPEBGXZ
+@ cdecl -arch=arm ?c_str@?$_Yarn@_W@std@@QBAPB_WXZ(ptr) msvcp140.?c_str@?$_Yarn@_W@std@@QBAPB_WXZ
+@ thiscall -arch=i386 ?c_str@?$_Yarn@_W@std@@QBEPB_WXZ(ptr) msvcp140.?c_str@?$_Yarn@_W@std@@QBEPB_WXZ
+@ cdecl -arch=win64 ?c_str@?$_Yarn@_W@std@@QEBAPEB_WXZ(ptr) msvcp140.?c_str@?$_Yarn@_W@std@@QEBAPEB_WXZ
 @ extern ?cerr@std@@3V?$basic_ostream@DU?$char_traits@D@std@@@1@A msvcp140.?cerr@std@@3V?$basic_ostream@DU?$char_traits@D@std@@@1@A
 @ extern ?cin@std@@3V?$basic_istream@DU?$char_traits@D@std@@@1@A msvcp140.?cin@std@@3V?$basic_istream@DU?$char_traits@D@std@@@1@A
 @ cdecl -arch=win32 ?classic@locale@std@@SAABV12@XZ() msvcp140.?classic@locale@std@@SAABV12@XZ
@@ -2248,12 +2248,12 @@
 @ cdecl -arch=arm ?empty@?$_Yarn@D@std@@QBA_NXZ(ptr) msvcp140.?empty@?$_Yarn@D@std@@QBA_NXZ
 @ thiscall -arch=i386 ?empty@?$_Yarn@D@std@@QBE_NXZ(ptr) msvcp140.?empty@?$_Yarn@D@std@@QBE_NXZ
 @ cdecl -arch=win64 ?empty@?$_Yarn@D@std@@QEBA_NXZ(ptr) msvcp140.?empty@?$_Yarn@D@std@@QEBA_NXZ
-@ stub -arch=arm ?empty@?$_Yarn@G@std@@QBA_NXZ
-@ stub -arch=i386 ?empty@?$_Yarn@G@std@@QBE_NXZ
-@ stub -arch=win64 ?empty@?$_Yarn@G@std@@QEBA_NXZ
-@ stub -arch=arm ?empty@?$_Yarn@_W@std@@QBA_NXZ
-@ stub -arch=i386 ?empty@?$_Yarn@_W@std@@QBE_NXZ
-@ stub -arch=win64 ?empty@?$_Yarn@_W@std@@QEBA_NXZ
+@ cdecl -arch=arm ?empty@?$_Yarn@G@std@@QBA_NXZ(ptr) msvcp140.?empty@?$_Yarn@G@std@@QBA_NXZ
+@ thiscall -arch=i386 ?empty@?$_Yarn@G@std@@QBE_NXZ(ptr) msvcp140.?empty@?$_Yarn@G@std@@QBE_NXZ
+@ cdecl -arch=win64 ?empty@?$_Yarn@G@std@@QEBA_NXZ(ptr) msvcp140.?empty@?$_Yarn@G@std@@QEBA_NXZ
+@ cdecl -arch=arm ?empty@?$_Yarn@_W@std@@QBA_NXZ(ptr) msvcp140.?empty@?$_Yarn@_W@std@@QBA_NXZ
+@ thiscall -arch=i386 ?empty@?$_Yarn@_W@std@@QBE_NXZ(ptr) msvcp140.?empty@?$_Yarn@_W@std@@QBE_NXZ
+@ cdecl -arch=win64 ?empty@?$_Yarn@_W@std@@QEBA_NXZ(ptr) msvcp140.?empty@?$_Yarn@_W@std@@QEBA_NXZ
 @ cdecl ?empty@locale@std@@SA?AV12@XZ(ptr) msvcp140.?empty@locale@std@@SA?AV12@XZ
 @ cdecl -arch=arm ?encoding@codecvt_base@std@@QBAHXZ(ptr) msvcp140.?encoding@codecvt_base@std@@QBAHXZ
 @ thiscall -arch=i386 ?encoding@codecvt_base@std@@QBEHXZ(ptr) msvcp140.?encoding@codecvt_base@std@@QBEHXZ
